---
title: "Setup for marker paper"
author: "Amanda Charbonneau"
date: "December 9, 2014"
output:
  html_document:
    toc: yes
    toc_depth: 3
---

This script takes nine datasets from varying years and sources, and reformats them into one large csv file that can be used to analyze flowering time. Some data sets have extra datapoints for traits other than flowering time, which are not transfered into the final document.

###Datasets

CultivarGH2013.csv

IsraelSpainPops2013GH.csv

LaleField2005.csv

2012FieldData.csv

2003QstParents.csv

2004QstOffspring.csv

2013plantsSpring.csv

Summer2010dataSummary.csv

```{r Settings, echo=FALSE, message=FALSE}
rm( list=ls())

getwd()

# Install function for packages    
packages<-function(x){
  x<-as.character(match.call()[[2]])
  if (!require(x,character.only=TRUE)){
    install.packages(pkgs=x,repos="http://cran.r-project.org")
    require(x,character.only=TRUE)
  }
}
packages(RColorBrewer)
packages(dplyr)
packages(plyr)

```

```{r ReadIN, echo=FALSE}

cult_2013 <- read.csv("../OriginalData/CultivarGH2013.csv", na.strings=".")

IS_2013 <- read.csv("../OriginalData/IsraelSpainPops2013GH.csv", na.strings=".")

lale2005 <- read.csv("../OriginalData/LaleField2005.csv", na.strings=".")

plants2012 <- read.csv("../OriginalData/2012FieldData.csv", sep=",", na.strings="")

qst_2003 <- read.csv("../OriginalData/2003QstParents.csv", na.strings=".")

qst_2004 <- read.csv("../OriginalData/2004QstOffspring.csv", na.strings=".")

spring2013 <- read.csv("../OriginalData/2013plantsSpring.csv", sep=",", na.strings="")

sum2010 <- read.csv("../OriginalData/Summer2010dataSummary.csv", na.strings=".")

Genotypes <- read.csv("../OriginalData/GenotypicData.csv", sep = ",", na.strings = c("NA", "", "."))

```

###Data Munging
####Fix dates for each dataset so they are all the same format, as in:
```{r Dates1}
# fix dates for Cultivar GH 2013.csv
cult_2013$planted <- as.POSIXct( strptime(cult_2013$planted, format="%m/%d/%y"))
cult_2013$Germ <- as.POSIXct( strptime(cult_2013$Germ, format="%m/%d/%y"))
cult_2013$crownTop <- as.POSIXct( strptime(cult_2013$crownTop, format="%m/%d/%y"))
cult_2013$crownSide <- as.POSIXct( strptime(cult_2013$crownSide, format="%m/%d/%y"))
cult_2013$FlowerDate <- as.POSIXct( strptime(cult_2013$FlowerDate, format="%m/%d/%y"))
cult_2013$FlowerPics <- as.POSIXct( strptime(cult_2013$FlowerPics, format="%m/%d/%y"))
cult_2013$TissueColl <- as.POSIXct( strptime(cult_2013$TissueColl, format="%m/%d/%y"))
cult_2013$ExCode <- "G1_13"

```

```{r Dates2, echo=FALSE}

# fix dates for IsraelSpain Pops 2013 GH.csv
IS_2013$Planted <- as.POSIXct( strptime( IS_2013$Planted, format="%m/%d/%y"))
IS_2013$Germ <- as.POSIXct( strptime( IS_2013$Germ, format="%m/%d/%y"))
IS_2013$crownTop <- as.POSIXct( strptime( IS_2013$crownTop, format="%m/%d/%y"))
IS_2013$crownSide <- as.POSIXct( strptime( IS_2013$crownSide, format="%m/%d/%y"))
IS_2013$X1stFlwrDate <- as.POSIXct( strptime( IS_2013$X1stFlwrDate, format="%m/%d/%y"))
IS_2013$FlwrPicDate <- as.POSIXct( strptime( IS_2013$FlwrPicDate, format="%m/%d/%y"))
IS_2013$Tissue.coll <- as.POSIXct( strptime( IS_2013$Tissue.coll, format="%m/%d/%y"))
IS_2013$GTF <- IS_2013$X1stFlwrDate - IS_2013$Germ
IS_2013$ExCode <- "G2_13"

# fix dates for LaleField2005.csv
lale2005$Planting_Date <- as.POSIXct( strptime(lale2005$Planting_Date, format="%m/%d/%y"))
lale2005$Germination_Date <- as.POSIXct( strptime(lale2005$Germination_Date, format="%m/%d/%y"))
lale2005$First_FlowerDate <- as.POSIXct( strptime(lale2005$First_FlowerDate, format="%m/%d/%y"))
lale2005$Mortality_Date <- as.POSIXct( strptime(lale2005$Mortality_Date, format="%m/%d/%y"))
lale2005$OvuleDate <- as.POSIXct( strptime(lale2005$OvuleDate, format="%m/%d/%y"))
lale2005$family <- as.factor(lale2005$family)
lale2005$PTF <- lale2005$First_FlowerDate - lale2005$Planting_Date
lale2005$ExCode <- "F_05"

# fix dates for 2012FieldData.csv
plants2012$Planting_Date <- as.POSIXct( strptime(plants2012$Planting_Date, format="%m/%d/%y"))
plants2012$Germ_Date <- as.POSIXct( strptime(plants2012$Germ_Date, format="%m/%d/%y"))
plants2012$Died_By <- as.POSIXct( strptime(plants2012$Died_By, format="%m/%d/%y"))
plants2012$Bolted_Date <- as.POSIXct( strptime(plants2012$Bolted_Date, format="%m/%d/%y"))
plants2012$Flowering_Date <- as.POSIXct( strptime(plants2012$Flowering_Date, format="%m/%d/%y"))
plants2012$Date_Blossom_Collected <- as.POSIXct( strptime(plants2012$Date_Blossom_Collected, format="%m/%d/%y"))
plants2012$second_bolt <- as.POSIXct( strptime(plants2012$second_bolt, format="%m/%d/%y"))
plants2012$SpringAliveDate <- as.POSIXct( strptime(plants2012$SpringAliveDate, format="%m/%d/%y"))
plants2012 <- droplevels(plants2012[plants2012$subpop != "unknown" & plants2012$subpop != "cross", ])
plants2012$PTF <- plants2012$Flowering_Date - plants2012$Planting_Date
plants2012$Survive_winter <- ifelse(plants2012$Survive_winter==0, "No", "Yes")
plants2012$ExCode <- "F_12"

# fix dates for 2003QstParents.csv"
qst_2003$PlantingDate <- as.POSIXct( strptime( qst_2003$PlantingDate, format="%m/%d/%y"))
qst_2003$ExCode <- "G_03"

# fix dates for 2004 QstOffspring.csv
qst_2004$plantingDate <- as.POSIXct( strptime( qst_2004$plantingDate, format="%d-%b-%y"))
qst_2004$ExCode <- "G_04"

# fix dates for 2013plantsSpring.csv
spring2013$PD <- as.POSIXct( strptime(spring2013$Plant_Date, format="%m/%d/%y"))
spring2013$GD <- as.POSIXct( strptime(spring2013$Germ_Date, format="%m/%d/%y"))
spring2013$BD <- as.POSIXct( strptime(spring2013$Bolt_Date, format="%m/%d/%y"))
spring2013$FD <- as.POSIXct( strptime(spring2013$Flower_Date, format="%m/%d/%y"))
spring2013$blossom <- as.POSIXct( strptime(spring2013$Blossom_Collected, format="%m/%d/%y"))
spring2013$PTF <- spring2013$FD - spring2013$PD
spring2013$DTB <- spring2013$BD - spring2013$GD
spring2013$DTF <- spring2013$FD - spring2013$GD
spring2013$ExCode <- "F_13"

# fix dates for Summer2010dataSummary.csv
sum2010$planted <- as.POSIXct( strptime( sum2010$planted, format="%m/%d/%y"))
sum2010$germDate <- as.POSIXct( strptime( sum2010$germDate, format="%m/%d/%y"))
sum2010$rosettePhoto <- as.POSIXct( strptime( sum2010$rosettePhoto, format="%m/%d/%y"))
sum2010$X1stflwr <- as.POSIXct( strptime( sum2010$X1stflwr, format="%m/%d/%y"))
sum2010$blossomPhoto <- as.POSIXct( strptime( sum2010$blossomPhoto, format="%m/%d/%y"))
sum2010$tissueColl <- as.POSIXct( strptime( sum2010$tissueColl, format="%m/%d/%y"))
sum2010$X2ndRosettePhoto <- as.POSIXct( strptime( sum2010$X2ndRosettePhoto, format="%m/%d/%y"))
sum2010$ExCode <- "G_10"
```
####Get subsets of data from the various datasets into the same format, with columns as follows:

**Experiment** is the name of the dataset the measurments came from

**ID** is a unique identifier (at least within Experiment) for that individual

**GrowthEnvironment** is where most of the plants obervations took place. Plants labeled "Field" may have been started in the greenhouse

**Pop** is the four letter population abbreviation for that individual

**PTF** is the number of whole days from planting (sowing seed) to the first open flower

**DTB** is the number of whole days from germination to bolting

**GTF** is the number of whole days from germination to the first open flower

**Year** is the calendar year in which the seeds were sown

**FlowDate** is the calendar date that the first flower opened

**PlantDate** is the calendar date that the seed was first planted

**GermData** is the calendar date that the seedling first emerged from the soil

**HeightFFcm** is the height from the soil to the base of the stem of the first open flower

**Vernalized** is whether that individual plant was put through a cold treatment

**DaysVern** is the total number of days that the individual was kept below 12.8C

**TimesVern** is the number of times that individual was cold treated

**Flowered** is whether that plant ever flowered

**FlowWoVern** is whether the plant flowered without vernalization

**Seedstock** is whether this individuals parents were grown in the field, greenhouse or commercially

**GermLocation** is whether this individual was started in the greenhouse, or in the field

**Geography** indicates (regionally) where in the world the stock was collected. East/West refer to regions within the Mediterranean

**Taxonomy** indicates species and subspecies for wild plants, and broad cultivar types for crops

**SubSpecies** indicates species and subspecies designations

**Species** indicates just species designations, ignoring subspecies divisions

**Habitat** is cultivar type for crops, and a controlled vocabulary description of theregion that the seeds were orignally collected from for wild plants

**SpecificOrigin** is the state/country/supply company each seed stock is from

**RegionOrigin** is the state/country/type of cultivar for each seed stock

**DOYP** is the calendar day (1-365) of year that the plant germinated, as experiments were started at different times of year and so experianced differing growth conditions like length of day and light intensity. We have planting day for all experiments, but know that in many cases there is up to a three month gap between planting and germination

**DOYG** is the calendar day (1-365) of year that the plant germinated. We don't have germination date for all experiments

**DOYGP** is the germination date if known, and the planting date if germination date is unknown

**DaysVernNoNA** is whether an individual was vernalized (x = 0 = no), and for how many days (x > 0 = # days)

**SurviveGTF** is usually the the number of days from germination to flowering, or bolting the following spring. Otherwise, if plant survived to the final fall observation, it is the number of days from germination to the final fall observation plus one. Or, if the plant survived until the following spring, but didn't flower by the final spring observation, it is the number of days from germination to the final spring observation plus one.

As in:
```{r Subsetting1}
columnnames <- c("Experiment", "ExCode", "ID", "GrowthEnvironment", "Pop", "PTF", "DTB", "GTF", "Year", "FlowDate", "PlantDate", "GermDate", "HeightFFcm", "Vernalized", "DaysVern", "TimesVern", "SurviveGTF")

# subset cult_2013
cult_2013sub <- data.frame(rep("Cult2013", length(cult_2013$Cultivar)),  	#Experiment
                         cult_2013$ExCode, #Experiment Code
                         paste(cult_2013$Cultivar, cult_2013$Pot, sep=""),		#ID
                         rep("greenhouse", length(cult_2013$Cultivar)), 		#GrowthEnvironment
                         cult_2013$Cultivar, 		#Pop
                         cult_2013$FlwrTime,		#PTF
                         rep(NA, length(cult_2013$Cultivar)), 		#DTB
                         round(cult_2013$FlowerDate - cult_2013$Germ), 		#GTF
                         rep("2013", length(cult_2013$Cultivar)), 		#Year
                         cult_2013$FlowerDate, 		#FlowDate
                         cult_2013$planted,     #PlantDate
                         cult_2013$Germ,		#GermDate
                         cult_2013$X1stFlwrHt_cm,		#HeightFFcm
                         rep("No", length(cult_2013$Cultivar)),		#Vernalized
                         rep(NA, length(cult_2013$Cultivar)),		#DaysVern
                         rep(NA, length(cult_2013$Cultivar)),		#TimesVern
                         ifelse(is.na(cult_2013$FlwrTime) == T, (as.POSIXct("12/2/13", format="%m/%d/%y") - cult_2013$Germ) + 1, 
                                round(cult_2013$FlowerDate - cult_2013$Germ)))   #SurviveGTF
```
#####Vernalization Data
For greenhouse studies, the columns **Vernalized**, **DaysVern** and **TimesVern** were treatment variables and were taken directly from the data sheets. For the three field seasons, plants did experiance vernalization, but as winter rather than as a dedicated treatment. For these studies, I based their vernalization on the maximum daily temperature of the city they were grown in. Variables for these three studies were determined in the following way:

**Vernalized** = Yes if plants survived the winter

**Vernalized times** = 1 for all field plants that survived the winter

**Vernalized length** = number of days between when max daily temp started staying below 12.8 to first day started staying above 12.8 based on data from http://www.almanac.com/weather/history/MI/Lansing/ (or Augusta)


> for i in `cat dates.txt`; do curl http://www.almanac.com/weather/history/MI/Lansing/${i} | grep "Maximum" >> Weatherdata/maxtemps.txt;done

I opened each of these files in TextWrangler, then did the following grep-based Find & Replace:

`find: .*/Lansing/(\d+-\d+-\d+)".*Maximum Temperature</h3><p><span class="value" > (\d+\.\d+).*`  

`replace: \1,\2`

For each resulting file, I imported it into R, and made a table that labeled each day that had a max temperature below 12.8 as "vern". Since these were all grown in Michigan, where daily temperatures in fall and spring fluxuate wildly, I chose the start and end dates as those where the temperature crossed the 12.8C threshold and then mostly stayed there. They're very subjective.

```{r Vernalization, echo=FALSE}
lale_05_06 <- read.csv(file = "../Weatherdata/maxtemps_05_06.txt", header=F)
lale_05_06$year <- substr( as.character(lale_05_06$V1), 1, 4 )
lale_05_06$vern <- ifelse(lale_05_06$V2 < 55, "vern", "")


winter_12_13 <- read.csv("../Weatherdata/maxtemps_12_13.txt", header=F)
winter_12_13$year <- substr( as.character(winter_12_13$V1), 1, 4 )
winter_12_13$vern <- ifelse(winter_12_13$V2 < 55, "vern", "")


winter_13_14 <- read.csv("../Weatherdata/maxtemps_13_14.txt", header=F)
winter_13_14$year <- substr( as.character(winter_13_14$V1), 1, 4 )
winter_13_14$vern <- ifelse(winter_13_14$V2 < 55, "vern", "")


```

```{r Subsetting2, echo=FALSE}
IS_2013sub <- data.frame(rep("IS2013", length(IS_2013$Population)),		#Experiment
                         IS_2013$ExCode, #Experiment Code
                         paste(IS_2013$Population, round(IS_2013$Pot), sep=""),		#ID
                         rep("greenhouse", length(IS_2013$Population)), 		#GrowthEnvironment
                         IS_2013$Population, 		#Pop
                         IS_2013$FlwrT,		#PTF
                         rep(NA, length(IS_2013$Population)), 		#DTB
                         IS_2013$GTF, 		#GTF
                         rep("2013", length(IS_2013$Population)), 		#Year
                         IS_2013$X1stFlwrDate, 		#FlowDate
                         IS_2013$Planted,      #PlantDate
                         IS_2013$Germ,		#GermDate
                         (IS_2013$X1stFlwrHt_mm / 100),		#HeightFFcm
                         rep("No", length(IS_2013$Population)),		#Vernalized
                         rep(NA, length(IS_2013$Population)),		#DaysVern
                         rep(NA, length(IS_2013$Population)),		#TimesVern
                         IS_2013$GTF)  #SurviveGTF
                                                  
lalesub <- data.frame(rep("Lale05", length(lale2005$population)),		#Experiment
                      lale2005$ExCode, #Experiment Code
                      paste(lale2005$family, lale2005$sibling, sep=""),		#ID
                      rep("field", length(lale2005$population)), 		#GrowthEnvironment
                      lale2005$population,		#Pop
                      lale2005$PTF, 		#PTF
                      rep(NA, length(lale2005$population)), 		#DTB
                      lale2005$DaysGermToFlwr, 		#GTF
                      rep("2005", length(lale2005$population)), 		#Year
                      lale2005$First_FlowerDate,		#FlowDate
                      lale2005$Planting_Date,      #PlantDate
                      lale2005$Germination_Date,		#GermDate
                      (lale2005$firstFlwrHt_mm / 100),		#HeightFFcm
                      ifelse(lale2005$First_FlowerDate > as.POSIXct("12/31/05", format="%m/%d/%y"), "Yes", "No"),		#Vernalized
                      ifelse(lale2005$First_FlowerDate > as.POSIXct("12/31/05", format="%m/%d/%y"), 150, NA),  #Nov 16- Apr 15		#DaysVern
                      ifelse(lale2005$First_FlowerDate > as.POSIXct("12/31/05", format="%m/%d/%y"), 1, NA),		#TimesVern
                      ifelse(is.na(lale2005$DaysGermToFlwr) == T & is.na(lale2005$OvuleDate) == F, NA,    
                          ifelse(is.na(lale2005$DaysGermToFlwr) == T & lale2005$Mortality_Date > as.POSIXct("11/29/05", format="%m/%d/%y"), 
                                 (lale2005$Mortality_Date + 86400) - lale2005$Germination_Date, 
                                  lale2005$DaysGermToFlwr))) #SurviveGTF
	
p2012sub <- data.frame(rep("field2012", length(plants2012$Code)),		#Experiment
                       plants2012$ExCode, #Experiment Code
                       paste(plants2012$Row, plants2012$Column, sep=""),		#ID
                       rep("field", length(plants2012$Code)),		#GrowthEnvironment
                       plants2012$Code,		#Pop
                       plants2012$PTF, 		#PTF
                       plants2012$Days_Germ_to_Bolt, 		#DTB
                       plants2012$Days_Germ_to_Flow, 		#GTF
                       rep("2012", length(plants2012$Code)), 		#Year
                       plants2012$Flowering_Date, 		#FlowDate
                       plants2012$Planting_Date,     #PlantDate
                       plants2012$Germ_Date,		#GermDate
                       plants2012$Height_1st_Flower_cm,		#HeightFFcm
                       ifelse(plants2012$Flowering_Date > as.POSIXct("12/31/12", format="%m/%d/%y"), "Yes", "No"),		#Vernalized
                       ifelse(plants2012$Flowering_Date > as.POSIXct("12/31/12", format="%m/%d/%y"), 168, NA), #Oct 29 - Apr 15		#DaysVern
                       ifelse(plants2012$Survive_winter=="No", NA, 1),		#TimesVern
                       
                       ifelse(is.na(plants2012$Died_By) == F & plants2012$Died_before_flowering == 1, NA, #Did you die in summer?, Yes = NA
                            ifelse(is.na(plants2012$Days_Germ_to_Flow) == F, plants2012$Days_Germ_to_Flow, #Did you flower by fall? Yes = GTF
                                  ifelse(is.na(plants2012$SpringAliveDate) == T, (as.POSIXct("9/12/12", format="%m/%d/%y") - plants2012$Germ_Date) + 1, #Did you die of winter Yes= FallDate
                                        #ifelse(is.na(plants2012$Bolted_Date) == F, (plants2012$Bolted_Date - plants2012$Germ_Date) + 1, #Did you bolt in spring? Yes = BoltDate
                            (as.POSIXct("5/28/13", format="%m/%d/%y") - plants2012$Germ_Date) + 1 ))))#) #Lived but didn't bolt by spring Yes = SpringDate
 #SurviveGTF


qst2003sub <- data.frame(rep("qst2003", length(qst_2003$Pop)),		#Experiment
                         qst_2003$ExCode, #Experiment Code
                         as.factor(qst_2003$UniqueID),		#ID
                         rep("greenhouse", length(qst_2003$Pop)),		#GrowthEnvironment
                         qst_2003$Pop,		#Pop
                         qst_2003$PlantToFlower,		#PTF
                         rep(NA, length(qst_2003$Pop)),		#DTB
                         qst_2003$GermToFlwr,		#GTF
                         rep("2003", length(qst_2003$Pop)),		#Year
                         (qst_2003$PlantingDate + (qst_2003$PlantToFlower * 60^2 * 24)),		#FlowDate
                         qst_2003$PlantingDate,    #PlantDate
                         (qst_2003$PlantingDate + (qst_2003$DaysToGerm * 60^2 * 24)),		#GermDate
                         qst_2003$X1stFlwrHgt_cm,		#HeightFFcm
                         ifelse(qst_2003$NumVernalz > 0, "Yes", "No"),		#Vernalized
                         qst_2003$DaysVernalized,		#DaysVern
                         qst_2003$NumVernalz,		#TimesVern
                         qst_2003$GermToFlwr)   #SurviveGTF

qst2004sub <- data.frame(rep("qst2004", length(qst_2004$Pop)),		#Experiment
                         qst_2004$ExCode, #Experiment Code
                         as.factor(qst_2004$ID),		#ID
                         rep("greenhouse", length(qst_2004$Pop)),		#GrowthEnvironment
                         qst_2004$Pop,		#Pop
                         qst_2004$plant2Flwr,		#PTF
                         rep(NA, length(qst_2004$Pop)),		#DTB
                         qst_2004$germ2flwr,		#GTF
                         rep("2004", length(qst_2004$Pop)),		#Year
                         (qst_2004$plantingDate + (qst_2004$plant2Flwr * 60^2 * 24)),		#FlowDate
                         qst_2004$plantingDate,    #PlantDate
                         (qst_2004$plantingDate + (qst_2004$germTime * 60^2 * 24)),		#GermDate
                         qst_2004$X1stFlwrHgt_cm,		#HeightFFcm
                         ifelse(qst_2004$NumVernal > 0, "Yes", "No"),		#Vernalized
                         qst_2004$DaysVernal,		#DaysVern
                         qst_2004$NumVernal,		#TimesVern
                         qst_2004$germ2flwr)   #SurviveGTF


s2013sub <- data.frame(rep("field2013", length(spring2013$Name)),		#Experiment
                       spring2013$ExCode, #Experiment Code
                       paste(spring2013$NS_Row, spring2013$WE_Row, sep=""),		#ID
                       rep("field", length(spring2013$Name)), 		#GrowthEnvironment
                       spring2013$Name, 		#Pop
                       spring2013$PTF,		#PTF
                       spring2013$DTB, 		#DTB
                       spring2013$DTF, 		#GTF
                       rep("2013", length(spring2013$Name)), 		#Year
                       spring2013$FD, 		#FlowDate
                       spring2013$PD,     #PlantDate
                       spring2013$GD,		#GermDate
                       spring2013$Height_1st_Flower_cm,		#HeightFFcm
                       rep("No", length(spring2013$Name)), # Oct 24 - Apr 6		#Vernalized
                       rep(NA, length(spring2013$Name)),		#DaysVern
                       rep(NA, length(spring2013$Name)),		#TimesVern
                       ifelse(is.na(spring2013$Died) == F & is.na(spring2013$Flower_Date) == T, NA,
                            ifelse(is.na(spring2013$DTF) == F, spring2013$DTF,
                                   round(as.POSIXct("11/14/13", format="%m/%d/%y") - spring2013$GD) + 1 ))) #SurviveGTF

sum2010sub <- data.frame(rep("sum2010", length(sum2010$Pop)),		#Experiment
                         sum2010$ExCode, #Experiment Code
                         paste(sum2010$Pop, sum2010$Plant, sep=""),		#ID
                         rep("greenhouse", length(sum2010$Pop)),		#GrowthEnvironment
                         sum2010$Pop,		#Pop
                         sum2010$DaysToFlwr,		#PTF
                         rep(NA, length(sum2010$Pop)),		#DTB
                         round(sum2010$X1stflwr - sum2010$germDate),		#GTF
                         rep("2010", length(sum2010$Pop)),		#Year
                         sum2010$X1stflwr,		#FlowDate
                         sum2010$planted,     #PlantDate
                         sum2010$germDate,		#GermDate
                         rep(NA, length(sum2010$Pop)),		#HeightFFcm
                         ifelse(sum2010$DaysVernal > 0, "Yes", "No"),		#Vernalized
                         sum2010$DaysVernal,		#DaysVern
                         sum2010$NumVernaliz,		#TimesVern
                         round(sum2010$X1stflwr - sum2010$germDate)) #SurviveGTF
```
####Datatype edits
I ran each subsetted dataset through a custom function that changed all named variables to factors, created several of the "Yes"/"No" columns, rounded numbers in calculations, and assigned **Seedstock** and **GermLocation** data before combining all the datasets into a single dataframe.

```{r Parsing, echo=FALSE}
ParseData <- function( mydata ) {
  names( mydata ) <- columnnames
  mydata$Pop <- factor( mydata$Pop)
  mydata$Pop <- as.factor(as.character(droplevels(mydata$Pop)))
  mydata$ExCode <- factor( mydata$ExCode)
  mydata$Flowered <- ifelse(mydata$PTF > 0, "Yes", "No")
  mydata$Flowered[is.na(mydata$Flowered)] <- "No"
  mydata$FlowWoVern <- ifelse(mydata$PTF > 0 & mydata$Vernalized == "No", "Yes",
                              ifelse(mydata$PTF > 0 & mydata$Vernalized == "Yes", "No", NA))
  mydata$PTF <- round(mydata$PTF)
  mydata$DTB <- round(mydata$DTB)
  mydata$GTF <- round(mydata$GTF)
  mydata$SurviveGTF <- round(mydata$SurviveGTF)
  mydata$Seedstock <- ifelse(  (mydata$Experiment == "IS2013" & mydata$Pop == "MAES") | 
                                       (mydata$Experiment == "Lale05" & mydata$Pop != "PBFR") |
                                       (mydata$Experiment == "qst2004"), "Greenhouse",      
                                          "Field")
  mydata$GermLocation <-   as.factor(ifelse(mydata$Experiment == "gh2006" | mydata$Experiment == "field2013", "InField", "InGH" ))

  mydata
}
cult_2013sub <- ParseData( cult_2013sub )
IS_2013sub <- ParseData(IS_2013sub)
lalesub <- ParseData(lalesub)
p2012sub <- ParseData(p2012sub)
qst2003sub <- ParseData(qst2003sub)
qst2004sub <- ParseData(qst2004sub)
s2013sub <- ParseData(s2013sub)
sum2010sub <- ParseData(sum2010sub)
```

```{r Combine, echo=FALSE}
CombinedDataSet <- data.frame(rbind(cult_2013sub, IS_2013sub, lalesub, p2012sub, qst2003sub, qst2004sub, s2013sub, sum2010sub))
```
####Adding Biological Information
To create the **Subspecies**, **Species**, **Geography**, **Taxonomy**, **Habitat**, **SpecificOrigin** and **RegionOrigin** columns, I made a series of assignment lists based on population names, as in:
```{r Assign1, echo=TRUE}
SubSpeciesList <- c(ADOL="sativus", AROL="sativus", CBBG="sativus", COOL="sativus", DAJO="sativus", ESNK="sativus", MABG="sativus", MYJO="sativus", NELO="sativus", OIBG="sativus", RABG="sativus", RACA="sativus", SPEU="sativus", TOBG="sativus", WMBG="sativus", AFFR="raphanistrum", DEES="raphanistrum", GHIL="raphanistrum", HCES="raphanistrum", HMES="raphanistrum", HZIL="raphanistrum", IMES="raphanistrum", MAES="raphanistrum", ZYIL="raphanistrum", AUFI="raphanistrum", COAU="raphanistrum", KAMI="raphanistrum", PBFR="landra", SAES="landra", AL="raphanistrum", BINY="raphanistrum", CBES="landra", GMIL="pugioniformis", M3AU="raphanistrum", N3="raphanistrum", NZIL="confusus", PG6="raphanistrum", REIL="raphanistrum", MAFI="raphanistrum", NAAU="raphanistrum", WEAU="raphanistrum", CGBC="sativus", FGBC="sativus", FRSI="sativus", LBBC="sativus", MBBC="sativus", NTJO="sativus", PABS="sativus", RABS="sativus", RBBC="sativus", SPNK="sativus", TYIL="raphanistrum")
                    
```

```{r Assign2, echo=FALSE}

SpeciesList <- c(ADOL="sativus", AROL="sativus", CBBG="sativus", COOL="sativus", DAJO="sativus", ESNK="sativus", MABG="sativus", MYJO="sativus", NELO="sativus", OIBG="sativus", RABG="sativus", RACA="sativus", SPEU="sativus", TOBG="sativus", WMBG="sativus", AFFR="raphanistrum", DEES="raphanistrum", GHIL="raphanistrum", HCES="raphanistrum", HMES="raphanistrum", HZIL="raphanistrum", IMES="raphanistrum", MAES="raphanistrum", ZYIL="raphanistrum", AUFI="raphanistrum", COAU="raphanistrum", KAMI="raphanistrum", PBFR="raphanistrum", SAES="raphanistrum",  AL="raphanistrum", BINY="raphanistrum", CBES="raphanistrum", GMIL="pugioniformis", M3AU="raphanistrum", N3="raphanistrum", NZIL="confusus", PG6="raphanistrum", REIL="raphanistrum", MAFI="raphanistrum", NAAU="raphanistrum", WEAU="raphanistrum", CGBC="sativus", FGBC="sativus", FRSI="sativus", LBBC="sativus", MBBC="sativus", NTJO="sativus", PABS="sativus", RABS="sativus", RBBC="sativus", SPNK="sativus", TYIL="raphanistrum")

GeographyList <- c(ADOL="NA", AROL="NA", CBBG="NA", COOL="NA", DAJO="NA", ESNK="NA", MABG="NA", MYJO="NA", NELO="NA", OIBG="NA", RABG="NA", RACA="NA", SPEU="NA", TOBG="NA", WMBG="NA", AFFR="west", DEES="west", GHIL="east", HCES="west", HMES="west", HZIL="east", IMES="west", MAES="west", ZYIL="east", AUFI="nonNative", COAU="nonNative", KAMI="nonNative", PBFR="west", SAES="west", AL="nonNative", BINY="nonNative",CBES="west", GMIL="pugioniformis", M3AU="nonNative", N3="nonNative", NZIL="confusus", PG6="nonNative", REIL="east", MAFI="nonNative", NAAU="nonNative", WEAU="nonNative", CGBC="NA", FGBC="NA", FRSI="NA", LBBC="NA", MBBC="NA", NTJO="NA", PABS="NA", RABS="NA", RBBC="NA", SPNK="NA", TYIL="east")

TaxonomyList <- c(ADOL="oleifera", AROL="oleifera", CBBG="european", COOL="oleifera", DAJO="european", ESNK="european", MABG="caudatus", MYJO="daikon", NELO="daikon", OIBG="oleifera", RABG="caudatus", RACA="caudatus", SPEU="european", TOBG="daikon", WMBG="daikon", AFFR="raphanistrum", DEES="raphanistrum", GHIL="raphanistrum", HCES="raphanistrum", HMES="raphanistrum", HZIL="raphanistrum", IMES="raphanistrum", MAES="raphanistrum", ZYIL="raphanistrum", AUFI="raphanistrum", COAU="raphanistrum", KAMI="raphanistrum", PBFR="landra", SAES="landra", AL="raphanistrum", BINY="raphanistrum", CBES="landra", GMIL="pugioniformis", M3AU="raphanistrum", N3="raphanistrum", NZIL="confusus", PG6="raphanistrum", REIL="raphanistrum", MAFI="raphanistrum", NAAU="raphanistrum", WEAU="raphanistrum", CGBC="daikon", FGBC="daikon", FRSI="european", LBBC="european", MBBC="european", NTJO="european", PABS="european", RABS="european", RBBC="european", SPNK="european", TYIL="raphanistrum")

HabitatList <- c(ADOL="cultivar", AROL="cultivar", CBBG="cultivar", COOL="cultivar", DAJO="cultivar", ESNK="cultivar", MABG="cultivar", MYJO="cultivar", NELO="cultivar", OIBG="cultivar", RABG="cultivar", RACA="cultivar", SPEU="cultivar", TOBG="cultivar", WMBG="cultivar", AFFR="agricultural", DEES="natural", GHIL="disturbed", HCES="disturbed", HMES="disturbed", HZIL="disturbed", IMES="disturbed", MAES="disturbed", ZYIL="disturbed", AUFI="agricultural", COAU="agricultural", KAMI="agricultural", PBFR="natural", SAES="natural",  AL="agricultural", BINY="agricultural", CBES="natural", GMIL="natural", M3AU="agricultural", N3="agricultural", NZIL="natural", PG6="agricultural", REIL="disturbed", MAFI="disturbed", NAAU="agricultural", WEAU="agricultural", CGBC="cultivar", FGBC="cultivar", FRSI="cultivar", LBBC="cultivar", MBBC="cultivar", NTJO="cultivar", PABS="cultivar", RABS="cultivar", RBBC="cultivar", SPNK="cultivar", TYIL="disturbed")

SpecificOriginList <- c(ADOL="MSU", AROL="MSU", CBBG="BountifulGardens", COOL="MSU", DAJO="JohnScheepers", ESNK="NKLawn", MABG="BountifulGardens", MYJO="JohnScheepers", NELO="JohnScheepers", OIBG="BountifulGardens", RABG="BountifulGardens", RACA="California", SPEU="NKLawn", TOBG="BountifulGardens", WMBG="BountifulGardens", AFFR="France", DEES="Spain", GHIL="Israel", HCES="Spain", HMES="Spain", HZIL="Israel", IMES="Spain", MAES="Spain", ZYIL="Israel", AUFI="Finland", COAU="Australia", KAMI="Michigan", PBFR="France", SAES="Spain",  AL="Australia", BINY="NewYork", CBES="Spain", GMIL="Israel", M3AU="Australia", N3="Australia", NZIL="Israel", PG6="Australia", REIL="Israel", MAFI="Finland", NAAU="Australia", WEAU="Australia", CGBC="BakersCreek", FGBC="BakersCreek", FRSI="SeedsItaly", LBBC="BakersCreek", MBBC="BakersCreek", NTJO="JohnScheepers", PABS="Burpees", RABS="Burpees", RBBC="BakersCreek", SPNK="NKLawn", TYIL="Israel")

RegionOriginList <- c(ADOL="oleifera", AROL="oleifera", CBBG="european", COOL="oleifera", DAJO="european", ESNK="european", MABG="caudatus", MYJO="daikon", NELO="daikon", OIBG="oleifera", RABG="caudatus", RACA="caudatus", SPEU="european", TOBG="daikon", WMBG="daikon", AFFR="France", DEES="Spain", GHIL="Israel", HCES="Spain", HMES="Spain", HZIL="Israel", IMES="Spain", MAES="Spain", ZYIL="Israel", AUFI="Finland", COAU="Australia", KAMI="Michigan", PBFR="France", SAES="Spain", AL="Australia", BINY="NewYork", CBES="Spain", GMIL="Israel", M3AU="Australia", N3="Australia", NZIL="Israel", PG6="Australia", REIL="Israel", MAFI="Finland", NAAU="Australia", WEAU="Australia", CGBC="daikon", FGBC="daikon", FRSI="european", LBBC="european", MBBC="european", NTJO="european", PABS="european", RABS="european", RBBC="european", SPNK="european", TYIL="Israel")

LocationList <- c(ADOL="oleifera", AROL="oleifera", CBBG="european", COOL="oleifera", DAJO="european", ESNK="european", MABG="caudatus", MYJO="daikon", NELO="daikon", OIBG="oleifera", RABG="caudatus", RACA="caudatus", SPEU="european", TOBG="daikon", WMBG="daikon", AFFR="raphNatW", DEES="raphNatW", GHIL="raphNatE", HCES="raphNatW", HMES="raphNatW", HZIL="raphNatE", IMES="raphNatW", MAES="raphNatW", ZYIL="raphNatE", AUFI="raphNN", COAU="raphNN", KAMI="raphNN", PBFR="lanmar", SAES="lanmar",  AL="raphNN", BINY="raphNN", CBES="lanmar", GMIL="pugioniformis", M3AU="raphNN", N3="raphNN", NZIL="none", PG6="raphNN", REIL="raphNatE", MAFI="raphNN", NAAU="raphNN", WEAU="raphNN", CGBC="daikon", FGBC="daikon", FRSI="european", LBBC="european", MBBC="european", NTJO="european", PABS="european", RABS="european", RBBC="european", SPNK="european", TYIL="raphNatE")

OrderList <- c(ADOL=18, AROL=19, CBBG=7, COOL=20, DAJO=8, ESNK=9, MABG=22, MYJO=3, NELO=4,OIBG=21, RABG=23, RACA=24, SPEU=17, TOBG=5, WMBG=6, AFFR=4, DEES=5, GHIL=10, HCES=6, HMES=7, HZIL=11, IMES=8, MAES=9, ZYIL=14, AUFI=1, COAU=3, KAMI=4, PBFR=1, SAES=3,  AL=7, BINY=2, CBES=2, GMIL=15, M3AU=7, N3=7, NZIL=16, PG6=7, REIL=12, MAFI=5, NAAU=6, WEAU=7, CGBC=1,FGBC=2, FRSI=10, LBBC=11, MBBC=12, NTJO=13, PABS=14, RABS=15, RBBC=16,SPNK=17, TYIL=13) 

```

```{r Assign3, echo=FALSE}
#This section uses the assignment lists created above to make new columns based on CombinedDataSet$Pop, and calculates the day of year (1-365) for planting and germination day

CombinedDataSet$Geography <- as.factor(GeographyList[CombinedDataSet$Pop])
CombinedDataSet$Taxonomy <- as.factor(TaxonomyList[CombinedDataSet$Pop])
CombinedDataSet$SubSpecies <- SubSpeciesList[CombinedDataSet$Pop]
CombinedDataSet$Species <- as.factor(SpeciesList[CombinedDataSet$Pop])
CombinedDataSet$Habitat <- as.factor(HabitatList[CombinedDataSet$Pop])
CombinedDataSet$SpecificOrigin <- as.factor(SpecificOriginList[CombinedDataSet$Pop])
CombinedDataSet$RegionOrigin <- as.factor(RegionOriginList[CombinedDataSet$Pop])
CombinedDataSet$locals <- as.factor(LocationList[CombinedDataSet$Pop])
CombinedDataSet$Order <- as.factor(OrderList[CombinedDataSet$Pop])
CombinedDataSet$Order <- as.numeric(CombinedDataSet$Order)
CombinedDataSet$Seedstock[CombinedDataSet$SubSpecies == "sativus"] <- "Nursery"
CombinedDataSet$SubSpecies <- as.factor(CombinedDataSet$SubSpecies)
CombinedDataSet$Seedstock <- as.factor(CombinedDataSet$Seedstock)
CombinedDataSet$Flowered <- as.factor(CombinedDataSet$Flowered)
CombinedDataSet$FlowWoVern <- as.factor(CombinedDataSet$FlowWoVern)
CombinedDataSet$DOYP <- strftime(CombinedDataSet$PlantDate, format = "%j")
CombinedDataSet$DOYP <- as.numeric(CombinedDataSet$DOYP)
CombinedDataSet$DOYG <- strftime(CombinedDataSet$GermDate, format = "%j")
CombinedDataSet$DOYG <- as.numeric(CombinedDataSet$DOYG)
CombinedDataSet$DOYGP <- ifelse(is.na(CombinedDataSet$DOYG), CombinedDataSet$DOYP, CombinedDataSet$DOYG)
CombinedDataSet$DaysVernNoNA <- CombinedDataSet$DaysVern
CombinedDataSet$DaysVernNoNA[is.na(CombinedDataSet$DaysVernNoNA)] <- 0
CombinedDataSet$FlowDate <- strftime(CombinedDataSet$FlowDate, format="%Y-%m-%d" )
CombinedDataSet$SurviveGTF <- round(CombinedDataSet$SurviveGTF)
CombinedDataSet$PTG <- round(CombinedDataSet$GermDate - CombinedDataSet$PlantDate)

CombinedDataSet <- droplevels(CombinedDataSet[CombinedDataSet$Pop != "N3",])

#SPEU is now SPNK; NELO now NEJS; RACA now RAJS. -JKC 
#Don't know which ID the three Austrailan pops really were

CombinedDataSet$Pop <- revalue(CombinedDataSet$Pop, c(
                               "M3AU" = "UKAU", 
                               "AL"   = "UKAU", 
                               "PG6"  = "UKAU",
                               "SPEU" = "SPNK",
                               "NELO" = "NEJS",
                               "RACA" = "RAJS"
                               ))



CombinedDataSet <- droplevels(CombinedDataSet)

```

```{r splitGenotypes, echo=FALSE}

JustMeta <- Genotypes[, c(1:2)]
JustSNPs <- Genotypes[,c(1:15)]
JustSSR <- Genotypes[c(1, 16:31)]

write.csv(JustSSR, "../MungedData/JustSSR.csv", row.names = F, quote=F)
system("python 1.1.2_biallele.py ../MungedData/JustSSR.csv ../MungedData/BiSSR.csv")

JustSSRFix <- read.csv("../MungedData/BiSSR.csv", header = T)
colnames(JustSSRFix) <- c("SSR.ID", colnames(JustSSRFix[2:109]))

BialleleGene <- dplyr::full_join( JustSNPs, JustSSRFix )

#Loop puts SNPs in format for STRUCTURE
# Genotype 0-> 1 1
# Genotype 1-> 1 2
# Genotype 2-> 2 2
for( i in 3:15 ) {
  colN <- ncol(JustMeta)
  
  JustMeta[, colN + 1] <- ifelse( JustSNPs[,i]==0, 1, ifelse( JustSNPs[,i]==1, 1, ifelse( JustSNPs[,i]==2, 2, -9 )))
  
  colnames(JustMeta)[ colN + 1] <- colnames(JustSNPs[i])
  
  JustMeta[, colN + 2 ] <- ifelse( JustSNPs[,i]==0, 1, ifelse( JustSNPs[,i]==1, 2, ifelse( JustSNPs[,i]==2, 2, -9 )))
  
    colnames(JustMeta)[ colN + 2] <- paste(colnames(JustSNPs[i]), "_2", sep="")
}


```


```{r write, echo=FALSE}
write.csv(CombinedDataSet, "../MungedData/CombinedDataSet.csv", quote=F, row.names=F)

write.csv(BialleleGene, "../MungedData/AllBialleleGeno.csv", row.names = F)

#SmartPCA Files

write.table(t(BialleleGene[, c(7:ncol(BialleleGene))]), "../MungedData/Marker.geno", row.names = F, col.names = F, sep="", na = "9")

write.table(cbind(BialleleGene$SSR.ID, rep("U", length(BialleleGene$Pop)), BialleleGene$Pop), "../MungedData/Marker.ind", row.names = F, quote = F, col.names = F, sep = "\t")

write.table(cbind( colnames(BialleleGene)[ 7:131 ], 
       c(rep(1,3), rep(2,3), rep(6,3), 8, 4, 9, 9, 
         rep(5,40), rep(7,16), rep(10,8), rep(1, 25), 
         rep(3,9), rep(6,14)), rep(0, 125), rep(0, 125)),
       "../MungedData/Marker.snp", 
       row.names = F, quote = F, col.names = F, sep = "\t")

#STRUCTURE files


STR_geno <- full_join(JustMeta, JustSSR)

STR_geno$Pop <- as.numeric(STR_geno$Pop)

STR_columns <- colnames(STR_geno)[seq(from = 3, to = 44, by=2)]

for (i in 1:20) {
  STR_geno_shuffle <- STR_geno[ sample( nrow( STR_geno ), nrow( STR_geno ), FALSE ) ,]
  write.table(c(STR_columns, "\n"), c(paste( "../MungedData/", i, "rand.txt", coll='', sep='')), 
              quote = FALSE, sep = " ", row.names = FALSE, eol = " ", col.names = FALSE)
  write.table( STR_geno_shuffle, paste( "../MungedData/", i, "rand.txt", coll='', sep=''), 
               quote = FALSE, sep = " ", row.names = FALSE, col.names = FALSE, append = TRUE, na = "-9")
}

```

###Error Checking
####Data Integrity
```{r Structure, echo=FALSE}
str(CombinedDataSet)

```

```{r Colors, echo=FALSE}

col_pal1 = brewer.pal(12, "Set3")
col_pal2 = brewer.pal(8, "Dark2")
col_pal3 = brewer.pal(12, "Paired")
col_pal = c(col_pal1, col_pal2, col_pal3)

```

Levels of each factor:
```{r Levels, echo=FALSE}
length(levels(CombinedDataSet$Pop))
levels(CombinedDataSet$Geography)
levels(CombinedDataSet$Taxonomy)
levels(CombinedDataSet$SubSpecies)
levels(CombinedDataSet$Species)
levels(CombinedDataSet$Habitat)
levels(CombinedDataSet$SpecificOrigin)
levels(CombinedDataSet$RegionOrigin)

```

```{r Tables, echo=FALSE}
table(CombinedDataSet$Pop)
table(CombinedDataSet$SubSpecies)
table(CombinedDataSet$Experiment)
```

####Plots of pivitol data
Note that the "SurviveGTF" data has a new spike in individuals that flowered around 170 days, which
is caused by creating flowering dates for plants that didn't flower, but were healthy before they 
froze to death

```{r, echo=FALSE}
par(mfrow=c(1,1))
SGTF <- hist(as.numeric(CombinedDataSet$SurviveGTF), breaks=50, plot = F)

hist(as.numeric(CombinedDataSet$SurviveGTF), breaks=SGTF$breaks, main = "SurviveGTF vs. GTF", col = rgb(1,0,0, alpha = .3), xlab="DTF")
hist(as.numeric(CombinedDataSet$GTF), breaks=SGTF$breaks, add=T, col = rgb(0,0,1, alpha = .3))
legend(400, 800, legend = c("SurviveGTF", "GTF"), pch=16, col = c(rgb(1,0,0, alpha = .3), rgb(0,0,1, alpha = .3)))


#plot(as.factor(FlowDate[Experiment == "field2013"]))


```

Plot showing latency of germination by experiment. Most experiments had several individuals that 
took a very long time to germinate, so using "Planting to Flowering" in the model instead of 
"Germination to Flowering" is going to cause serious issues.

```{r, echo=FALSE}
#germination latency
plot(CombinedDataSet$PTG, pch=16, col=col_pal3[CombinedDataSet$Experiment], main = "Planting to Germination")
legend(550, 150, legend = levels(CombinedDataSet$Experiment), ncol = 3, bty = "n", col=col_pal3[], pch = 16)

```

Boxplot of days from planting to flowering of each population

```{r Plots1, echo=FALSE}
par(las=2, cex.axis=.7)
plot(CombinedDataSet$Pop, CombinedDataSet$PTF)
```

Barplot showing number of individuals per population

```{r Plots2, echo=FALSE}
par(las=2, cex.axis=.6)
plot(CombinedDataSet$Pop)
```

