---
title: "Phenotypic Analysis"
author: "Amanda Charbonneau"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document: 
    theme: "cerulean"
    number_sections: true
    toc: true
    toc_depth: 5
    toc_float: true
    collapsed: false
    df_print: paged
    code_folding: hide
---

```{r SourcePackages, include=FALSE}

# Install function for packages    
packages<-function(x){
  x<-as.character(match.call()[[2]])
  if (!require(x,character.only=TRUE)){
    install.packages(pkgs=x,repos="http://cran.r-project.org")
    require(x,character.only=TRUE)
  }
}

packages( plotrix ) #shading
packages( cowplot ) #ggplot gridding
packages( lme4 ) #modeling
packages( MCMCglmm ) #modeling
packages( ggthemes )
packages( ggplot2 ) #plotting
packages( reshape2 ) #munging
packages( dplyr ) #munging
packages( car ) #anova

hmcols <- colorRampPalette(c("black", "white"))(100)

#Standard Error of a proportion
SEP <- function( x, n ){
  B <- sqrt( ( x * (1-x) ) / (n - 1) )
}

```


```{r ImportData, echo=FALSE}
FlowData <- read.csv( "../MungedData/CombinedDataSet.csv" )
FlowData$FlowDate <- as.Date( FlowData$FlowDate )
FlowData$GermDate <- as.Date( FlowData$GermDate )
FlowData$PlantDate <- as.Date( FlowData$PlantDate )

#Standardize flowering dates
FlowData$DOYPS <- ifelse( FlowData$DOYP < 78, FlowData$DOYP + 287, FlowData$DOYP - 78 )
FlowData$DOYGS <- ifelse( FlowData$DOYG < 78, FlowData$DOYG + 287, FlowData$DOYG - 78 )

#mean standardize numeric predictors
FlowData$msDOYPS <- FlowData$DOYPS - mean(FlowData$DOYPS, na.rm=T)
FlowData$msDOYGS <- FlowData$DOYGS - mean(FlowData$DOYGS, na.rm=T)
FlowData$msDaysVernNoNA <- FlowData$DaysVernNoNA - mean(FlowData$DaysVernNoNA, na.rm=T)

FlowData$BoltFlower <- ifelse(is.na(FlowData$GTF) == F,  FlowData$GTF, FlowData$DTB)

#Dropping PABS, MBBC, NTJO, Confusus and UKAU
FlowData <- filter(FlowData, SubSpecies !="confusus" & Pop !="UKAU" &
                   Pop !="PABS" & Pop !="MBBC" & Pop !="NTJO")

FlowData <- droplevels(FlowData)



```

Global parameters for MCMCglmm

```{r ModelParameters}

nitt <- 1000000   # Number of iterations
thin <- 100       # Thinning interval
burnin <- 10000   # burnin period
pr <- TRUE       # save posterior distribution of random effects

```




```{r RrrSubset, echo=FALSE}
# Data Subset for R.r. raphanistrum analysis
#Only keep data for individuals that are Rrr, have a flowering date and germination date
RrrData <- filter( FlowData, Taxonomy == "raphanistrum" & 
                             !is.na( SurviveGTF) & 
                             !is.na( msDOYGS )  ) %>% 
           select( ID, GTF, Geography, Seedstock, GrowthEnvironment, msDOYGS, 
                   msDaysVernNoNA, Pop, Experiment, Habitat, PTF, msDOYPS, PTG, 
                   SurviveGTF, FlowWoVern)
    

RrrData$Geography <- relevel( RrrData$Geography, "west" )
RrrData$Habitat <- relevel( RrrData$Habitat, "natural" )
RrrData$Ag <- RrrData$Habitat
levels(RrrData$Ag) <- c("nonag", "ag", "nonag", "nonag" )
RrrData <- droplevels( RrrData )

```

```{r RrNativeSubset, echo=FALSE}
# Data Subset for Raphanus natives analysis
#Only keep data for individuals that are natives, have a flowering date and germination date
NatData <- filter( FlowData, Geography != "nonNative" & 
                             Species != "sativus" & 
                             !is.na( SurviveGTF) & 
                             !is.na( msDOYGS )  ) %>% 
           select( ID, GTF, Geography, Seedstock, GrowthEnvironment, msDOYGS, 
                   msDaysVernNoNA, Pop, Experiment, Habitat, PTF, msDOYPS, 
                   Species, SubSpecies, Taxonomy, SurviveGTF, FlowWoVern)

NatData$ID <- row.names( NatData )
NatData$Geography <- relevel( NatData$Geography, "west" )
NatData$Habitat <- relevel( NatData$Habitat, "natural" )
NatData$SubSpecies <- relevel( NatData$SubSpecies, "raphanistrum" )
NatData <- droplevels( NatData )
```

```{r SativusSubset, echo=FALSE}
# Data Subset for sativus analysis
#Only keep data for individuals that are sativus, have a flowering date and germination date
SatData <- filter( FlowData, Species == "sativus" & 
                             !is.na( SurviveGTF) & 
                             !is.na( msDOYGS ) ) %>% 
           select( ID, GTF, Geography, Seedstock, GrowthEnvironment, msDOYGS, 
                   msDaysVernNoNA, Pop, Experiment, Habitat, PTF, msDOYPS, 
                   Species, SubSpecies, Taxonomy, SurviveGTF, FlowWoVern)

SatData$Taxonomy <- relevel(SatData$Taxonomy, "european")
SatData <- droplevels( SatData )
```

```{r JeffSativusSubset, echo=FALSE}
# Data Subset for sativus analysis
#Only keep data for individuals that are sativus, have a flowering date and germination date and were in the original data from Jeff
JeffSatData <- filter( FlowData, Species == "sativus" & 
                             !is.na( SurviveGTF) & 
                             !is.na( msDOYGS )  &
                             ( Pop == "MABG" |
                               Pop == "RABG" |
                               Pop == "RACA" |
                               Pop == "ADOL" |
                               Pop == "AROL" |
                               Pop == "COOL" |
                               Pop == "CBBG" |
                               Pop == "DAJO" |
                               Pop == "ESNK" |
                               Pop == "MYJO" |
                               Pop == "NELO" |
                               Pop == "SPNK" |
                               Pop == "TOBG" |
                               Pop == "WMBG" )) %>% 
           select( ID, GTF, Geography, Seedstock, GrowthEnvironment, msDOYGS, 
                   msDaysVernNoNA, Pop, Experiment, Habitat, PTF, msDOYPS, 
                   Species, SubSpecies, Taxonomy, SurviveGTF )

JeffSatData$Taxonomy <- relevel(JeffSatData$Taxonomy, "european")
JeffSatData <- droplevels( JeffSatData )
```

```{r RawSubset, echo=FALSE}
# Data Subset for raw data figure
#Only keep data for individuals that are in experiment, have a flowering date and germination date

RawData <- filter( FlowData, !is.na( BoltFlower ) & 
                             !is.na( msDOYGS )  ) %>% 
           select( ID, GTF, BoltFlower, Geography, Seedstock, GrowthEnvironment, msDOYGS, 
                   msDaysVernNoNA, Pop, Experiment, Habitat, PTF, msDOYPS, PTG, 
                   SurviveGTF, FlowWoVern, Taxonomy, Species, Order, locals, ExCode)
    

RawData$Geography <- relevel( RawData$Geography, "west" )
RawData$Habitat <- relevel( RawData$Habitat, "natural" )
RawData$Experiment <- relevel(RawData$Experiment, "field2013")
RawData$Ag <- RawData$Habitat
levels(RawData$Ag) <- c("nonag", "ag", "nonag", "nonag" )
RawData <- droplevels( RawData )

```

```{r ActualRawData, echo=FALSE}
# Data Subset for table of experiments
# Numbers for what was planned/done regardless of who died

ExtraRawData <- select( FlowData, ID, GTF, Geography, Seedstock, 
                   GrowthEnvironment, msDOYGS, msDaysVernNoNA, 
                   Pop, Experiment, Habitat, PTF, msDOYPS, PTG, 
                   SurviveGTF, FlowWoVern, Taxonomy, Species, 
                   Order, locals, ExCode)

ExtraRawData$Geography <- relevel( ExtraRawData$Geography, "west" )
ExtraRawData$Habitat <- relevel( ExtraRawData$Habitat, "natural" )
ExtraRawData$Ag <- ExtraRawData$Habitat
levels(ExtraRawData$Ag) <- c("nonag", "ag", "nonag", "nonag" )
ExtraRawData <- droplevels( ExtraRawData )


```

```{r unsubsettedmodeldata, echo=FALSE}
AllModeled <- filter( FlowData, !is.na( SurviveGTF) & 
                                !is.na( msDOYGS )  ) %>% 
           select( ID, GTF, Geography, Seedstock, GrowthEnvironment, msDOYGS, 
                   msDaysVernNoNA, Pop, Experiment, Habitat, PTF, msDOYPS, 
                   Species, SubSpecies, Taxonomy, SurviveGTF, FlowWoVern)

AllModeled <- droplevels( AllModeled )
```

#Summary Data

Numbers of planted individuals (experiments as planned)

Total individuals: `r length(ExtraRawData$Pop)`

Number of Pops: `r length(levels(ExtraRawData$Pop))`

Indivduals per Pop: `r (dplyr::count(ExtraRawData, Pop) %>% summarize(min=min(n)))$min` - `r (dplyr::count(ExtraRawData, Pop) %>% summarize(max=max(n)))$max`

Populations modeled, overall: `r levels(AllModeled$Pop)` N= `r length(levels(AllModeled$Pop))`

Individuals modeled, overall: `r length(AllModeled$ID)`

Populations in Raw figure: `r levels(RawData$Pop)` N= `r length(levels(RawData$Pop))`

Individuals in Raw figure: `r length(RawData$Pop)`

##Data Subset for R.r. raphanistrum analysis. 
Only kept data for individuals that are Rrr, have a flowering date and germination date.
This includes individuals that were given a +1 flowering date:

Total individuals: `r length(RrrData$Pop)`

Number of Pops: `r length(levels(RrrData$Pop))`

Indivduals per Pop: `r (dplyr::count(RrrData, Pop) %>% summarize(min=min(n)))$min` - `r (dplyr::count(RrrData, Pop) %>% summarize(max=max(n)))$max`

## Data Subset for Raphanus natives analysis
Only keep data for individuals that are natives, have a flowering date and germination date.
This includes individuals that were given a +1 flowering date:

Total individuals: `r length(NatData$Pop)`

Number of Pops: `r length(levels(NatData$Pop))`

Indivduals per Pop: `r (dplyr::count(NatData, Pop) %>% summarize(min=min(n)))$min` - `r (dplyr::count(NatData, Pop) %>% summarize(max=max(n)))$max`

## Data Subset for sativus analysis
Only keep data for individuals that are sativus, have a flowering date and germination date.
This includes individuals that were given a +1 flowering date:

Total individuals: `r length(SatData$Pop)`

Number of Pops: `r length(levels(SatData$Pop))`

Indivduals per Pop: `r (dplyr::count(SatData, Pop) %>% summarize(min=min(n)))$min` - `r (dplyr::count(SatData, Pop) %>% summarize(max=max(n)))$max`

## Raw numbers
Combined data from the above datasets but NOT including +1 flowering date. 
This was used to create the flowering boxplot figure

Total individuals: `r length(RawData$Pop)`

Number of Pops: `r length(levels(RawData$Pop))`

Indivduals per Pop: `r (dplyr::count(RawData, Pop) %>% summarize(min=min(n)))$min` - `r (dplyr::count(RawData, Pop) %>% summarize(max=max(n)))$max`

## Raw +1 numbers
Combined data from the datasets used for modeling, DOES include +1 flowering date. 

Total individuals: `r length(AllModeled$Pop)`

Number of Pops: `r length(levels(AllModeled$Pop))`

Indivduals per Pop: `r (dplyr::count(AllModeled, Pop) %>% summarize(min=min(n)))$min` - `r (dplyr::count(AllModeled, Pop) %>% summarize(max=max(n)))$max`



Pops per Experiment:
```{r PopExp, echo=FALSE}
dplyr::count( ExtraRawData, ExCode, Pop ) %>% dplyr::count( ExCode )

```

Individuals per Experiment: 

```{r IndExp, echo=FALSE}
dplyr::count( ExtraRawData, Experiment, ExCode )
```

Individuals per Pop, per Experiment:
```{r IndPopExp, echo=FALSE}
dplyr::count( ExtraRawData, ExCode, Pop ) %>% dplyr::group_by( ExCode ) %>% summarise(minN=min(n), maxN=max(n))
```

#Modeling

##Model for just *R.r. raphanistrum*

Response variable: Days from germination to flowering
Here, I'm using germination to flowering (or last alive date) as the response, which will be more accurate than planting to flowering, but have many more NA's.

```{r RrrModels, echo=TRUE, results="hide"}

Rrr_lmer <- lmer( SurviveGTF ~ Geography + Seedstock + msDOYGS + msDaysVernNoNA + (1|Experiment) + (1|Geography:Pop), data=RrrData  )

Rrr_glmm <- MCMCglmm( fixed = SurviveGTF ~ Geography  + Seedstock + msDOYGS + msDaysVernNoNA, 
                      random = ~ Experiment + Geography:Pop,
                      nitt = nitt,
                      thin = thin,
                      burnin = burnin,
                      pr = pr, #save posterior distribution of random effects
                      data = RrrData)

Rrr_glmm_noV <- MCMCglmm( fixed = SurviveGTF ~ Geography  + Seedstock + msDOYGS, 
                      random = ~ Experiment + Geography:Pop,
                      nitt = nitt,
                      thin = thin,
                      burnin = burnin,
                      pr = pr, #save posterior distribution of random effects
                      data = RrrData)

```

##Model for just native-range *Raphanus*

This is the model that is roughly equivalent to the model we used to look at the differences between the ***R.r. raphanistrum*** plants, just using **SubSpecies** instead of **Geography**. 

```{r RrNatModels, echo=TRUE, results="hide"}

Nat_lmer <- lmer( SurviveGTF ~ SubSpecies + Seedstock + msDOYGS + msDaysVernNoNA + (1|Experiment) + (1|SubSpecies:Pop), data = NatData  )


Nat_glmm <- MCMCglmm( fixed = SurviveGTF ~ SubSpecies + Seedstock + msDOYGS + msDaysVernNoNA, 
                      random = ~ Experiment + SubSpecies:Pop, 
                      nitt = nitt,
                      thin = thin,
                      burnin = burnin,
                      pr = pr, #save posterior distribution of random effects
                      data = NatData)

Nat_glmm_noV <- MCMCglmm( fixed = SurviveGTF ~ SubSpecies + Seedstock + msDOYGS, 
                      random = ~ Experiment + SubSpecies:Pop, 
                      nitt = nitt,
                      thin = thin,
                      burnin = burnin,
                      pr = pr, #save posterior distribution of random effects
                      data = NatData)
```

##Model for just crops
This is the model that is roughly equivalent to the model we used to look at the differences between the ***R.r. raphanistrum*** plants, just using **Variety** instead of **Geography**.

```{r SatModels, echo=TRUE, results="hide"}

Sat_lmer <- lmer( SurviveGTF ~ Taxonomy + (1|Experiment) + (1|Taxonomy:Pop), data=SatData  )


Sat_glmm <- MCMCglmm( fixed = SurviveGTF ~ Taxonomy, 
                      random = ~ Experiment + Taxonomy:Pop, 
                      nitt = nitt,
                      thin = thin,
                      burnin = burnin,
                      pr = pr, #save posterior distribution of random effects
                      data = SatData)


JeffSat_glmm <- MCMCglmm( fixed = SurviveGTF ~ Taxonomy, 
                      random = ~ Experiment + Taxonomy:Pop, 
                      nitt = nitt,
                      thin = thin,
                      burnin = burnin,
                      pr = pr, #save posterior distribution of random effects
                      data = JeffSatData)

```

###Model output for just *R.r. raphanistrum*
```{r ModelStatsRrr, echo=FALSE}
summary( Rrr_glmm )

```
###Model output for just *R.r. raphanistrum*, no vernalization predictor
```{r ModelStatsRrrNoV, echo=FALSE}

summary( Rrr_glmm_noV )
```

###Model output for just native range *R. raphanus*
```{r ModelStatsNat, echo=FALSE}

summary( Nat_glmm )

```
###Model output for just native range *R. raphanus*, no vernalization predictor
```{r ModelStatsNatNoV, echo=FALSE}
summary( Nat_glmm_noV )
```

###Model output for just *R.r. sativus*
```{r ModelStatsSat, echo=FALSE}
summary( Sat_glmm )

```

###Model output for just genotyped subset of *R.r. sativus*
```{r JeffModelStatsSat, echo=FALSE}
summary( JeffSat_glmm )

```


```{r RrrModelMeans, echo=FALSE}
Rrr_pop_chains <- Rrr_glmm$Sol[ , 14:31 ] + Rrr_glmm$Sol[ , 1 ]

# Calculating population level means

# east effect
Rrr_pop_chains[ ,c(6, 9, 15, 16, 18) ] <- Rrr_pop_chains[ ,c(6, 9, 15, 16, 18) ] + Rrr_glmm$Sol[, 2]

#nonNative effect
Rrr_pop_chains[ , c( 2, 3, 4, 11, 13, 14, 17 ) ] <- Rrr_pop_chains[ , c( 2, 3, 4, 11, 13, 14, 17 ) ] + Rrr_glmm$Sol[, 3]

Rrr_pop <- as.data.frame( cbind( colMeans( Rrr_pop_chains ), 
                          as.data.frame( HPDinterval( Rrr_pop_chains ) ) ))


Rrr_pop[,4:6] <- data.frame( matrix( 
                  unlist( strsplit( rownames( Rrr_pop ), split = ".", fixed = TRUE ) ),
                  nrow=18, byrow=T ) )

colnames(Rrr_pop) <- c( "Mean", "Lower", "Upper", "Group", "Geography", "Pop" )

Rrr_pop <- arrange( Rrr_pop, Geography )

# Calculating Geographic region level means

Rrr_fixed_chains <- mutate( data.frame( Rrr_glmm$Sol ),
                            west = X.Intercept., 
                            east = X.Intercept. + Geographyeast, 
                            nonNative = X.Intercept. + GeographynonNative ) %>%
                    select( west, east, nonNative )

Rrr_fixed <- data.frame( cbind( colMeans( Rrr_fixed_chains ), 
                         data.frame( HPDinterval( as.mcmc(Rrr_fixed_chains) ) ) ) )

Rrr_fixed <- mutate(Rrr_fixed, species = rownames(Rrr_fixed), Mean = colMeans.Rrr_fixed_chains.)

```


```{r NoV_RrrModelMeans, echo=FALSE}

# Calculating population level means

Rrr_noV_pop_chains <- Rrr_glmm_noV$Sol[ , 13:30 ] + Rrr_glmm_noV$Sol[ , 1 ]

# east effect
Rrr_noV_pop_chains[ ,c(6, 9, 15, 16, 18) ] <- Rrr_noV_pop_chains[ ,c(6, 9, 15, 16, 18) ] + Rrr_glmm_noV$Sol[, 2]

#nonNative effect
Rrr_noV_pop_chains[ , c( 2, 3, 4, 11, 13, 14, 17 ) ] <- Rrr_noV_pop_chains[ , c( 2, 3, 4, 11, 13, 14, 17 ) ] + Rrr_glmm_noV$Sol[, 3]

Rrr_noV_pop <- as.data.frame( cbind( colMeans( Rrr_noV_pop_chains ), 
                          as.data.frame( HPDinterval( Rrr_noV_pop_chains ) ) ))
Rrr_noV_pop[,4:6] <- data.frame( matrix( 
                  unlist( strsplit( rownames( Rrr_noV_pop ), split = ".", fixed = TRUE ) ),
                  nrow=18, byrow=T ) )
colnames(Rrr_noV_pop) <- c( "Mean", "Lower", "Upper", "Group", "Geography", "Pop" )
Rrr_noV_pop <- arrange( Rrr_noV_pop, Geography )

# Calculating Geographic region level means

Rrr_noV_fixed_chains <- mutate( data.frame( Rrr_glmm_noV$Sol ),
                            west = X.Intercept., 
                            east = X.Intercept. + Geographyeast, 
                            nonNative = X.Intercept. + GeographynonNative ) %>%
                    select( west, east, nonNative )

Rrr_noV_fixed <- data.frame( cbind( colMeans( Rrr_noV_fixed_chains ), 
                         data.frame( HPDinterval( as.mcmc(Rrr_noV_fixed_chains) ) ) ) )
Rrr_noV_fixed <- mutate(Rrr_noV_fixed, species = rownames(Rrr_noV_fixed), Mean = colMeans.Rrr_noV_fixed_chains.)

```


```{r NatModelMeans, echo=FALSE}

Nat_pop_chains <- Nat_glmm$Sol[ , 14:28 ] + Nat_glmm$Sol[ , 1 ]

# Calculating population level means


# landra effect
Nat_pop_chains[ , c( 2, 11, 13 ) ] <- Nat_pop_chains[ , c( 2, 11, 13 ) ] + Nat_glmm$Sol[, 2]

# pugioniformis effect
Nat_pop_chains[ , 5 ] <- Nat_pop_chains[ , 5 ] + Nat_glmm$Sol[, 3]


Nat_pop <- as.data.frame( cbind( colMeans( Nat_pop_chains ), 
                          as.data.frame( HPDinterval( Nat_pop_chains ) ) ))
Nat_pop[,4:6] <- data.frame( matrix( 
                  unlist( strsplit( rownames( Nat_pop ), split = ".", fixed = TRUE ) ),
                  nrow = 15, byrow = T ) )
colnames( Nat_pop ) <- c( "Mean", "Lower", "Upper", "Group", "SubSp", "Pop" )
Nat_pop <- arrange( Nat_pop, SubSp ) 

# Calculating Geographic region level means

Nat_fixed_chains <- mutate( data.frame( Nat_glmm$Sol ),
                            raphanistrum = X.Intercept., 
                            landra = X.Intercept. + 
                            SubSpecieslandra,
                            pugioniformis = X.Intercept. + SubSpeciespugioniformis ) %>%
                    select( raphanistrum, landra, pugioniformis)

Nat_fixed <- data.frame( cbind( colMeans( Nat_fixed_chains ), 
                         data.frame( HPDinterval( as.mcmc( Nat_fixed_chains ) ) ) ) )
Nat_fixed <- mutate(Nat_fixed, species = rownames(Nat_fixed), Mean = colMeans.Nat_fixed_chains.)


```


```{r NoV_NatModelMeans, echo=FALSE}

Nat_NoV_pop_chains <- Nat_glmm_noV$Sol[ , 13:27 ] + Nat_glmm_noV$Sol[ , 1 ]

# Calculating population level means

# landra effect
Nat_NoV_pop_chains[ , c( 2, 11, 13 ) ] <- Nat_NoV_pop_chains[ , c( 2, 11, 13 ) ] + Nat_glmm_noV$Sol[, 2]

# rostratus effect
Nat_NoV_pop_chains[ , 5 ] <- Nat_NoV_pop_chains[ , 5 ] + Nat_glmm_noV$Sol[, 3]


Nat_noV_pop <- as.data.frame( cbind( colMeans( Nat_NoV_pop_chains ), 
                          as.data.frame( HPDinterval( Nat_NoV_pop_chains ) ) ))
Nat_noV_pop[,4:6] <- data.frame( matrix( 
                  unlist( strsplit( rownames( Nat_noV_pop ), split = ".", fixed = TRUE ) ),
                  nrow = 15, byrow = T ) )
colnames( Nat_noV_pop ) <- c( "Mean", "Lower", "Upper", "Group", "SubSp", "Pop" )
Nat_noV_pop <- arrange( Nat_noV_pop, SubSp ) 

# Calculating Geographic region level means

Nat_NoV_fixed_chains <- mutate( data.frame( Nat_glmm_noV$Sol ),
                            raphanistrum = X.Intercept., 
                            landra = X.Intercept. + SubSpecieslandra,
                            pugioniformis = X.Intercept. + SubSpeciespugioniformis ) %>%
                    select( raphanistrum, landra, pugioniformis)

Nat_NoV_fixed <- data.frame( cbind( colMeans( Nat_NoV_fixed_chains ), 
                         data.frame( HPDinterval( as.mcmc( Nat_NoV_fixed_chains ) ) ) ) )
Nat_NoV_fixed <- mutate(Nat_NoV_fixed, species = rownames(Nat_NoV_fixed), Mean = colMeans.Nat_NoV_fixed_chains.)


```


```{r SatModelMeans, echo=FALSE}

Sat_pop_chains <- Sat_glmm$Sol[ , 8:28 ] + Sat_glmm$Sol[ , 1 ]

# Calculating population level means

# caudatus effect
Sat_pop_chains[ , c( 11, 15, 17 ) ] <- Sat_pop_chains[ , c( 11, 15, 17 ) ] + Sat_glmm$Sol[ , 2 ]

# daikon effect
Sat_pop_chains[ , c( 4, 8, 12, 13, 20, 21 ) ] <- Sat_pop_chains[ , c( 4, 8, 12, 13, 20, 21 ) ] + Sat_glmm$Sol[ , 3 ]

# oleifera effect
Sat_pop_chains[ , c( 1, 2, 5, 14 ) ] <- Sat_pop_chains[ , c( 1, 2, 5, 14 ) ] + Sat_glmm$Sol[ , 4 ]

Sat_pop <- as.data.frame( cbind( colMeans( Sat_pop_chains ), 
                          as.data.frame( HPDinterval( Sat_pop_chains ) ) ))
Sat_pop[ , 4:6 ] <- data.frame( matrix( 
                  unlist( strsplit( rownames( Sat_pop ), split = ".", fixed = TRUE ) ),
                  nrow = 21, byrow = T ) )

colnames( Sat_pop ) <- c( "Mean", "Lower", "Upper", "Group", "Variety", "Pop" )
levels( Sat_pop$Variety ) <- c( "Rat-tail", "Daikon", "European", "Oil-seed" )
Sat_pop <- arrange( Sat_pop, Variety )

# Calculating Geographic region level means

Sat_fixed_chains <- mutate( data.frame( Sat_glmm$Sol ),
                            european = X.Intercept.,
                            caudatus = X.Intercept. + Taxonomycaudatus,
                            daikon = X.Intercept. + Taxonomydaikon,
                            oleifera = X.Intercept. + Taxonomyoleifera ) %>%
                    select( european, caudatus, daikon, oleifera )

Sat_fixed <- data.frame( cbind( colMeans( Sat_fixed_chains ), 
                         data.frame( HPDinterval( as.mcmc( Sat_fixed_chains ) ) ) ) )
Sat_fixed <- mutate(Sat_fixed, species = rownames(Sat_fixed), Mean = colMeans.Sat_fixed_chains.)
```


```{r JeffSatModelMeans, echo=FALSE}

JeffSat_pop_chains <- JeffSat_glmm$Sol[ , 8:19 ] + JeffSat_glmm$Sol[ , 1 ]

# Calculating population level means

# caudatus effect
JeffSat_pop_chains[ , c( 7, 9) ] <- JeffSat_pop_chains[ , c( 7, 9 ) ] + JeffSat_glmm$Sol[ , 2 ]

# daikon effect
JeffSat_pop_chains[ , c( 8, 11, 12) ] <- JeffSat_pop_chains[ , c( 8, 11, 12 ) ] + JeffSat_glmm$Sol[ , 3 ]

# oleifera effect
JeffSat_pop_chains[ , c( 1, 2, 4 ) ] <- JeffSat_pop_chains[ , c( 1, 2, 4 ) ] + JeffSat_glmm$Sol[ , 4 ]

JeffSat_pop <- as.data.frame( cbind( colMeans( JeffSat_pop_chains ), 
                          as.data.frame( HPDinterval( JeffSat_pop_chains ) ) ))
JeffSat_pop[ ,4:6 ] <- data.frame( matrix( 
                  unlist( strsplit( rownames( JeffSat_pop ), split = ".", fixed = TRUE ) ),
                  nrow = 12, byrow = T ) )
colnames( JeffSat_pop ) <- c( "Mean", "Lower", "Upper", "Group", "Variety", "Pop" )
levels( JeffSat_pop$Variety ) <- c( "Rat-tail", "Daikon", "European", "Oil-seed" )
JeffSat_pop <- arrange( JeffSat_pop, Variety )

# Calculating Geographic region level means


JeffSat_fixed_chains <- mutate( data.frame( JeffSat_glmm$Sol ),
                            european = X.Intercept.,
                            caudatus = X.Intercept. + Taxonomycaudatus,
                            daikon = X.Intercept. + Taxonomydaikon,
                            oleifera = X.Intercept. + Taxonomyoleifera ) %>%
                    select( european, caudatus, daikon, oleifera )

JeffSat_fixed <- data.frame( cbind( colMeans( JeffSat_fixed_chains ), 
                         data.frame( HPDinterval( as.mcmc( JeffSat_fixed_chains ) ) ) ) )
JeffSat_fixed <- mutate(JeffSat_fixed, species = rownames(JeffSat_fixed), Mean = colMeans.JeffSat_fixed_chains.)
```


#Plotting

##Only Native range *R. raphanus*
```{r PlotNatPopMeans, echo=FALSE, fig.path="../Figures/", fig.keep="last", fig.width=11, eval=F, message=F, warning=FALSE}
plot.new()
Nat_overplot <- rbind( mutate(Nat_pop, Data = "Yes"), mutate( Nat_noV_pop, Data = "No"))

Nat_overplot$SubSp
levels(Nat_overplot$SubSp)[levels(Nat_overplot$SubSp)=="pugioniformis"] <- ""
levels(Nat_overplot$SubSp)[levels(Nat_overplot$SubSp)=="landra"] <- "R.r.landra"
levels(Nat_overplot$SubSp)[levels(Nat_overplot$SubSp)=="raphanistrum"] <- "R.r.raphanistrum"

NatPlot <- ggplot( Nat_overplot, aes( y = Mean, x = Pop, colour = Data ) ) + 
              geom_point( cex = 2.5) + 
              geom_errorbar( aes( y = Mean, x = Pop, max = Upper, min = Lower, colour = Data), width = 0, cex = 1) + 
              facet_grid( ~ SubSp, scales = "free", space = "free") +
              xlab( "Population" ) + 
              ylab( "Population means as \ndays from germination to flower" ) + 
              labs(color='Vernalization\nModeled') +
              scale_colour_grey(start = .7, end = .2) +
              ggtitle( expression( "Only Native-range"~italic( "Raphanus" ) ) ) 

NatPlot + theme_bw() + theme( axis.text.x = element_text( angle = -50, size = 14, hjust = -.05 ),
                                 axis.text.y = element_text( size = 14 ),
                                 axis.title.y = element_text( size = 14 ),
                                 strip.text.x = element_text( angle = 0, size = 9, face = "italic" ) ) 


mtext(text = expression(italic("R.pugioniformis")), side = 3, line = -.95, at=.21, cex = .8)
```

```{r PlotNatMeans, echo=FALSE, fig.path="../Figures/", fig.keep="last", warning=FALSE}

Nat_fixed_overplot <- rbind( mutate(Nat_fixed, Data = "Yes")[2:6], mutate( Nat_NoV_fixed, Data = "No")[2:6] )

NatPlot <- ggplot( Nat_fixed_overplot, aes(y = Mean, x = species, colour = Data)) + 
              geom_point( cex = 2.5) + 
              geom_errorbar( aes( y = Mean, max = upper, min = lower, colour = Data), width = 0, cex = 1) + 
              xlab( "Species or Sub-Species" ) + 
              ylab( "Means as \ndays from germination to flower" ) + 
              labs(color='Vernalization\nModeled') +
              scale_colour_grey(start = .7, end = .2) +
              ggtitle( expression( "Mean days to flower for all"~italic( "Raphanus" ) ) ) 

NatPlot + theme_bw() + theme( axis.text.x = element_text( angle = -50, size = 14, hjust = -.05, face = "italic" ),
                                 axis.text.y = element_text( size = 14 ),
                                 axis.title.y = element_text( size = 14 ),
                                 strip.text.x = element_text( angle = 0, size = 10, face = "italic" ) ) 


```

##All *R.r. raphanistrum*
```{r PlotRrrPopMeans, echo=FALSE, fig.path="../Figures/", fig.keep="last", warning=FALSE}
Rrr_pop_overplot <- rbind( mutate(Rrr_pop, Data = "Yes"), mutate( Rrr_noV_pop, Data = "No"))

Rrr_pop_overplot$Geography <- relevel(as.factor(Rrr_pop_overplot$Geography), ref="west")

RrrPlot <- ggplot( Rrr_pop_overplot, aes( x = Pop, y = Mean, col = Data ) ) + 
              geom_point( cex = 2.5 ) + 
              geom_errorbar( aes( y = Mean, max = Upper, min = Lower, col = Data), width = 0, cex = 1 ) +  
              facet_grid( ~Geography, scales = "free", space = "free" ) +
              xlab( "Population" ) + 
              ylab( "Population means as \ndays from germination to flower" ) +                       labs(color='Vernalization\nModeled') +
              scale_colour_grey(start = .7, end = .2) +
              ggtitle( expression( "All" ~ italic( "R.r. raphanistrum" ) ) ) 

RrrPlot + theme_bw() + theme( axis.text.x = element_text( angle = -50, size=14, hjust = -.05 ),
                                 axis.text.y = element_text( size = 14 ),
                                 axis.title.y = element_text( size = 14 ),
                                 strip.text.x = element_text( angle = 0, size = 10 ) ) 

```

```{r PlotRrrMeans, echo=FALSE, fig.path="../Figures/", fig.keep="last", warning=FALSE}
Rrr_fixed_overplot <- rbind( mutate(Rrr_fixed, Data = "Yes")[2:6], mutate( Rrr_noV_fixed, Data = "No")[2:6] )

Rrr_fixed_overplot$species <- relevel(as.factor(Rrr_fixed_overplot$species), ref="west")

RrrPlot <- ggplot( Rrr_fixed_overplot, aes( x = species, y = Mean, col = Data ) ) + 
              geom_point( cex = 2.5 ) + 
              geom_errorbar( aes( y = Mean, max = upper, min = lower, col = Data), width = 0, cex = 1 ) +  
              xlab( "Geographic Origin" ) + 
              ylab( "Means as \ndays from germination to flower" ) + 
              labs(color='Vernalization\nModeled') +
              scale_colour_grey(start = .7, end = .2) +
              ggtitle( expression( "Geographic means of" ~ italic( "R.r. raphanistrum" ) ) ) 

RrrPlot + theme_bw() + theme( axis.text.x = element_text( angle = -50, size=14, hjust = -.05 ),
                                 axis.text.y = element_text( size = 14 ),
                                 axis.title.y = element_text( size = 14 ),
                                 strip.text.x = element_text( angle = 0, size = 10 ) ) 

```

##Only crops *R. sativus*
```{r PlotSatPopMeans, echo=FALSE, fig.path="../Figures/", fig.keep="last", warning=FALSE}

SatPlot <- ggplot( Sat_pop, aes( x = Pop, y = Mean) ) + 
              geom_point( cex = 2.5 ) + 
              geom_errorbar( aes( y = Mean, max = Upper, min = Lower ), col = "gray40", width=0, cex=1 ) +  
              facet_grid( ~ Variety, scales = "free", space = "free" ) +
              xlab( "Population" ) + 
              ylab( "Population means as \ndays from germination to flower" ) + 
              ggtitle( expression( "Only crops" ~ ( italic( "R. sativus" ) ) ) ) 

SatPlot + theme_bw() + theme( axis.text.x = element_text( angle = -50, size = 14, hjust = -.05 ),
                                 axis.text.y = element_text( size = 14 ),
                                 axis.title.y = element_text( size = 14 ),
                                 strip.text.x = element_text( angle = 0, size = 10 ) )

```

```{r PlotSatMeans, echo=FALSE, fig.path="../Figures/", fig.keep="last", warning=FALSE}

SatPlot <- ggplot( Sat_fixed, aes( x = species, y = Mean) ) + 
              geom_point( cex = 2.5 ) + 
              geom_errorbar( aes( y = Mean, max = upper, min = lower ), col = "gray40", width=0, cex=1 ) +  
              xlab( "Crop variety" ) + 
              ylab( "Means as \ndays from germination to flower" ) + 
              ggtitle( expression( "Crop" ~ ( italic( "R. sativus" ) ) ~ "variety means" ) ) 

SatPlot + theme_bw() + theme( axis.text.x = element_text( angle = -50, size = 14, hjust = -.05 ),
                                 axis.text.y = element_text( size = 14 ),
                                 axis.title.y = element_text( size = 14 ),
                                 strip.text.x = element_text( angle = 0, size = 10 ) )

```

##Only crops genotyped subset *R. sativus*
```{r PlotJeffSatPopMeans, echo=FALSE, fig.path="../Figures/", fig.keep="last", warning=FALSE}

JeffSatPlot <- ggplot( JeffSat_pop, aes( x = Pop, y = Mean) ) + 
              geom_point( cex = 2.5 ) + 
              geom_errorbar( aes( y = Mean, max = Upper, min = Lower ), col = "gray40", width=0, cex=1 ) +  
              facet_grid( ~ Variety, scales = "free", space = "free" ) +
              xlab( "Population" ) + 
              ylab( "Population means as \ndays from germination to flower" ) + 
              ggtitle( expression( "Only crops" ~ ( italic( "R. sativus" ) ) ) ) 

JeffSatPlot + theme_bw() + theme( axis.text.x = element_text( angle = -50, size = 14, hjust = -.05 ),
                                 axis.text.y = element_text( size = 14 ),
                                 axis.title.y = element_text( size = 14 ),
                                 strip.text.x = element_text( angle = 0, size = 10 ) )

```

```{r PlotJeffSatMeans, echo=FALSE, fig.path="../Figures/", fig.keep="last", warning=FALSE}

JeffSatPlot <- ggplot( JeffSat_fixed, aes( x = species, y = Mean) ) + 
              geom_point( cex = 2.5 ) + 
              geom_errorbar( aes( y = Mean, max = upper, min = lower ), col = "gray40", width=0, cex=1 ) +  
              xlab( "Crop variety" ) + 
              ylab( "Means as \ndays from germination to flower" ) + 
              ggtitle( expression( "Crop" ~ ( italic( "R. sativus" ) ) ~ "variety means" ) ) 

JeffSatPlot + theme_bw() + theme( axis.text.x = element_text( angle = -50, size = 14, hjust = -.05 ),
                                 axis.text.y = element_text( size = 14 ),
                                 axis.title.y = element_text( size = 14 ),
                                 strip.text.x = element_text( angle = 0, size = 10 ) )

```

```{r TableofVern, echo=FALSE, warning=FALSE, message=FALSE, warning=FALSE}

# Building a table with percentage flowered without vernalization

PS_data <-  filter(FlowData, !is.na(SurviveGTF)) %>% 
              select( FlowWoVern, Pop) %>% 
              mutate( FlowWoVern = replace( FlowWoVern, is.na( FlowWoVern ) == T, "No")) 


PS_table <- data.frame( cbind( "No" = table(PS_data, useNA = "no")[1,],
                               "Yes" = table(PS_data, useNA = "no")[2,] ) ) 

MD_table <- select( FlowData, Pop, Taxonomy, Geography, SurviveGTF) %>%
            group_by( Taxonomy, Geography, Pop) %>%
            summarize(MeanGTF=mean(SurviveGTF, na.rm=T)) 


PS_table <- tbl_df( mutate( PS_table, PnoVern = ( Yes / (Yes + No) ) * 100,
                      N = Yes + No, 
                      Pop = factor(rownames(PS_table) ),
                      SEPnoVern = SEP( PnoVern/100, N )))

PS_table <- full_join( PS_table, MD_table, by="Pop" ) %>%
            mutate( FixGeography = replace( as.character(Geography), is.na( Geography ) == T, "Crop") )

PS_table$RPV <- ifelse(PS_table$PnoVern < 1, 1, round(PS_table$PnoVern) )

PS_table$N2 <- PS_table$N + 20

PS_table <- filter(PS_table, N > 0)

```


```{r RawDataSubsets, echo=FALSE, warning=FALSE}
native.data <- filter(RawData, locals=="pugioniformis" | locals=="raphNatW" |
                                   locals=="raphNatE") %>% arrange(Order)
natcov <- group_by(native.data, Pop, locals) %>% summarise(avg=mean(BoltFlower), stdd=sd(BoltFlower), min=min(BoltFlower), max=max(BoltFlower), COV=(stdd/avg * 100))

lanmar.data <- filter(RawData, locals=="lanmar") %>% arrange(Order)

lanmarcov <- group_by(lanmar.data, Pop, locals) %>% summarise(avg=mean(BoltFlower), stdd=sd(BoltFlower), min=min(BoltFlower), max=max(BoltFlower), COV=(stdd/avg * 100))

weed.data <- filter(RawData, locals=="raphNN") %>% arrange(Order)

weedcov <- group_by(weed.data, Pop, locals) %>% summarise(avg=mean(BoltFlower), stdd=sd(BoltFlower), min=min(BoltFlower), max=max(BoltFlower), COV=(stdd/avg * 100))

crop.data <- filter(RawData, Species=="sativus") %>% arrange(Order)

cropcov <- group_by(crop.data, Pop, locals) %>% summarise(avg=mean(BoltFlower), stdd=sd(BoltFlower), min=min(BoltFlower), max=max(BoltFlower), COV=(stdd/avg * 100))

ExpCoV <- group_by(RawData, Pop, Experiment, locals) %>% summarise(avg=mean(BoltFlower), stdd=sd(BoltFlower), min=min(BoltFlower), max=max(BoltFlower), COV=(stdd/avg * 100))

groupcov <- group_by(RawData, locals) %>% summarise(avg=mean(BoltFlower), stdd=sd(BoltFlower), min=min(BoltFlower), max=max(BoltFlower), COV=(stdd/avg * 100))

popcov <- group_by(RawData, Pop, locals) %>% summarise(avg=mean(BoltFlower), stdd=sd(BoltFlower), min=min(BoltFlower), max=max(BoltFlower), COV=(stdd/avg * 100))


write.csv(x = groupcov, file = "../Figures/GroupCoV.csv")
write.csv(x = ExpCoV, file = "../Figures/ExpCoV.csv")
write.csv(x = popcov, file = "../Figures/PopCoV.csv")

```


```{r RawDataPlot, echo=FALSE, fig.path="../Figures/", fig.keep="last", message=FALSE, warning=FALSE, fig.height=8, fig.width=11, warning=FALSE}

#Everything is plotted twice so that the axis lines go behind the plots 

# Plot landra
par(mfrow=c(1,1), mar=c(0,0,0,0), cex=.7, las=1, fig=c(0.1,.3,.69,.97))
boxplot(lanmar.data$BoltFlower ~ lanmar.data$Order, frame=F, axes=F, ylim=c(0,350), 
        varwidth=TRUE, at=c(1, 1.8, 2.7))

axis(side = 2, line=-1, tck=.95, col='gray85')

boxplot(lanmar.data$BoltFlower ~ lanmar.data$Order, add=T, frame=F, axes=F, ylim=c(0,350), 
        varwidth=TRUE, at=c(1, 1.8, 2.7),
        col=hmcols[c(PS_table$RPV[PS_table$Pop=="PBFR"],
                     PS_table$RPV[PS_table$Pop=="CBES"],
                     PS_table$RPV[PS_table$Pop=="SAES"])])

axis(side = 1, at=c(1, 1.8, 2.7), tick = F, line=-.6, labels=c("France\n(PBFR)","Spain\n(CBES)",
                                              "Spain\n(SAES)"))
par(cex=.9)
axis(side=3, at=2, labels=expression(italic("R.r. landra")), cex=2, tick=F, line=-1)

# Plot RRR

par(fig=c(0.3,1,.69,.97), cex=.7, new=TRUE)
boxplot(native.data$BoltFlower ~ native.data$Order, frame=F, axes=F, ylim=c(0,200),
        varwidth=TRUE, boxwex=1.1, col="white", border="white", at=c(seq(from=1, to=10.9, by=.9)))

axis(side = 2, line=-2.8, lty=0)

axis(side = 2, line=-2, tck=.9, col="gray85", labels = F)

boxplot(native.data$BoltFlower ~ native.data$Order, add=T, frame=F, axes=F, ylim=c(0,200),
        varwidth=TRUE, boxwex=1.1,  at=c(seq(from=1, to=10.9, by=.9)), col=hmcols[c(
        PS_table$RPV[PS_table$Pop=="AFFR"], PS_table$RPV[PS_table$Pop=="DEES"],
        PS_table$RPV[PS_table$Pop=="HCES"], PS_table$RPV[PS_table$Pop=="HMES"],
        PS_table$RPV[PS_table$Pop=="IMES"], PS_table$RPV[PS_table$Pop=="MAES"],
        PS_table$RPV[PS_table$Pop=="GHIL"], PS_table$RPV[PS_table$Pop=="HZIL"],
        PS_table$RPV[PS_table$Pop=="REIL"], PS_table$RPV[PS_table$Pop=="TYIL"],
        PS_table$RPV[PS_table$Pop=="ZYIL"], PS_table$RPV[PS_table$Pop=="GMIL"])])
        
axis(side = 1, at=c(seq(from=1, to=10.9, by=.9)), tick = F, line=-.8, labels=c( 
                                "France\n(AFFR)", "Spain\n(DEES)", 
                                "Spain\n(HCES)", "Spain\n(HMES)", 
                                "Spain\n(IMES)", "Spain\n(MAES)", 
                                "Israel\n(GHIL)", "Israel\n(HZIL)", 
                                "Israel\n(REIL)", "Israel\n(TYIL)",
                                "Israel\n(ZYIL)", "Israel\n(GMIL)"))



par(cex=.9)

axis(side=3, at=3.5, labels="Western", cex=2, tick=F, line=-3)
axis(side=3, at=8.5, labels="Eastern", cex=2, tick=F, line=-3)

axis(side=3, at=10.7, labels=expression(italic("R. pugioniformis")), cex=2, tick=F, line=-1)
axis(side=3, at=6, labels=expression(paste(italic("R.r. raphanistrum")," inside native range")), cex=1.2, tick=F, line=-1)

lines(c(6.1,6.1), c(50,150), lty=3)
lines(c(10.5,10.5), c(25,175), lty=6)

# Plot weeds

par(fig=c(0.1,.8,.37,.70), new=TRUE, cex=.7)
boxplot(weed.data$BoltFlower ~ weed.data$Order, frame=F, axes=F, ylim=c(0,200))
axis(side = 2, line=-1, tck=.9, col="gray85", yaxp=c(0, 150, 3))

boxplot(weed.data$BoltFlower ~ weed.data$Order, add=T, frame=F, axes=F, ylim=c(0,200),
        col=hmcols[c(
        PS_table$RPV[PS_table$Pop=="AUFI"], PS_table$RPV[PS_table$Pop=="BINY"],
        PS_table$RPV[PS_table$Pop=="COAU"], PS_table$RPV[PS_table$Pop=="KAMI"], 
        PS_table$RPV[PS_table$Pop=="MAFI"], PS_table$RPV[PS_table$Pop=="NAAU"],
        PS_table$RPV[PS_table$Pop=="WEAU"])],
        varwidth=TRUE)

axis(side = 1, at=c(1:7), tick = F, line=-1,
     label=c("Finland\n(AUFI)", "New York\n(BINY)",
             "Australia 1\n(COAU)", "Michigan\n(KAMI)",
             "Finland\n(MAFI)", "Australia 3\n(NAAU)",
             "Australia 2\n(WEAU)" ))


par(las=3)
mtext(text="Flowering Time in Days", side = 2, adj = 0, line = 2 )
par(las=1)

par(cex=1)
axis(side = 3, at=4.5, tick = F, line=-5, labels=expression(paste(italic("R.r. raphanistrum")," outside native range")))

par(fig=c(0.8,1,.4,.65), new=TRUE, cex=.7)


gradient.rect(4.5, 2, 5.5, 130, col=hmcols, gradient = "vertical", border=NA)
text(3, 150, labels = "Flowered without\ncold treatment", cex=1.2)
text(2.5, 10, labels="0% of population")
text(2.5, 120, labels="100% of population")
polygon(c(.4,6,6,.4), c(0,0,170,170))


par(fig=c(0.1,1,.03,.31), new=TRUE, cex=.7)

boxplot(crop.data$BoltFlower ~ crop.data$Order, frame=F, axes=F, ylim=c(0,200),
        varwidth=TRUE, boxwex=.7, at=c(seq(from=.9, to=18.9, by=.9)))
axis(side = 2, line=-1, tck=.95, col='gray85')
par(new=TRUE, cex=.7)

boxplot(crop.data$BoltFlower ~ crop.data$Order, add=T, frame=F, axes=F, ylim=c(0,200),
        varwidth=TRUE, boxwex=.7, at=c(seq(from=.9, to=18.9, by=.9)), col=hmcols[c(
        PS_table$RPV[PS_table$Pop=="CGBC"], PS_table$RPV[PS_table$Pop=="FGBC"],
        PS_table$RPV[PS_table$Pop=="MYJO"], PS_table$RPV[PS_table$Pop=="NEJS"],
        PS_table$RPV[PS_table$Pop=="TOBG"], PS_table$RPV[PS_table$Pop=="WMBG"],
        PS_table$RPV[PS_table$Pop=="CBBG"], PS_table$RPV[PS_table$Pop=="DAJO"],
        PS_table$RPV[PS_table$Pop=="ESNK"], PS_table$RPV[PS_table$Pop=="FRSI"],
        PS_table$RPV[PS_table$Pop=="LBBC"], PS_table$RPV[PS_table$Pop=="RABS"], 
        PS_table$RPV[PS_table$Pop=="RBBC"], PS_table$RPV[PS_table$Pop=="SPNK"], 
        PS_table$RPV[PS_table$Pop=="ADOL"], PS_table$RPV[PS_table$Pop=="AROL"], 
        PS_table$RPV[PS_table$Pop=="COOL"], PS_table$RPV[PS_table$Pop=="OIBG"], 
        PS_table$RPV[PS_table$Pop=="MABG"], PS_table$RPV[PS_table$Pop=="RABG"], 
        PS_table$RPV[PS_table$Pop=="RAJS"])])

axis(side = 1, at=c(seq(from=.9, to=19, by=1.8)), tick = F, line=-1.5, ylim=c(0, 300),
     label=c("CGBC","MYJO","TOBG","CBBG","ESNK",
             "LBBC","RBBC","ADOL","COOL","MABG","RAJS"))

axis(side = 1, at=c(seq(from=1.8, to=19, by=1.8)), tick = F, line=-1.5, ylim=c(0, 300),
     label=c("FGBC","NEJS","WMBG","DAJO","FRSI",
             "RABS","SPNK","AROL","OIBG","RABG"))

par(cex=1)
axis(side = 3, at=9.5, tick = F, line=-1, labels=expression(italic("R. sativus")))

par(cex=.9)
axis(side=3, at=2.7, tick=F, labels = "Daikon", line=-4)
axis(side=3, at=9.5, tick=F, labels="European", line=-4)
axis(side=3, at=15, tick=F, labels="Oilseed", line=-4)
axis(side=3, at=18, tick=F, labels="Rattail", line=-4)


lines(c(5.85,5.85), c(25,175), lty=6)
lines(c(13.1,13.1), c(25,175), lty=6)
lines(c(16.7,16.7), c(25,175), lty=6)




```

```{r RawDataDotPlot, echo=FALSE, fig.path="../Figures/", fig.keep="last", message=FALSE, warning=FALSE, fig.height=8, fig.width=11, warning=FALSE}

#Everything is plotted twice so that the axis lines go behind the plots 

# Plot landra
par(mfrow=c(1,1), mar=c(0,0,0,0), cex=.7, las=1, fig=c(0.1,.3,.69,.97))


plot(lanmar.data$GTF ~ lanmar.data$Order, add=T, frame=F, axes=F, ylim=c(0,350), 
         at=c(1, 1.8, 2.7), pch=16,
        bg=hmcols[c(PS_table$RPV[PS_table$Pop=="PBFR"],
                     PS_table$RPV[PS_table$Pop=="CBES"],
                     PS_table$RPV[PS_table$Pop=="SAES"])])
axis(side = 2, line=-1, tck=.95, bg='gray85')

axis(side = 1, at=c(1, 1.8, 2.7), tick = F, line=-.6, labels=c("France\n(PBFR)","Spain\n(CBES)",
                                              "Spain\n(SAES)"))
par(cex=.9)
axis(side=3, at=2, labels=expression(italic("R.r. landra")), cex=2, tick=F, line=-1)

# Plot RRR

par(fig=c(0.3,1,.69,.97), cex=.7, new=TRUE)


plot(native.data$GTF ~ native.data$Order, add=T, frame=F, axes=F, ylim=c(0,200), pch=16,
         wex=1.1,  at=c(seq(from=1, to=10.9, by=.9)), bg=hmcols[c(
        PS_table$RPV[PS_table$Pop=="AFFR"], PS_table$RPV[PS_table$Pop=="DEES"],
        PS_table$RPV[PS_table$Pop=="HCES"], PS_table$RPV[PS_table$Pop=="HMES"],
        PS_table$RPV[PS_table$Pop=="IMES"], PS_table$RPV[PS_table$Pop=="MAES"],
        PS_table$RPV[PS_table$Pop=="GHIL"], PS_table$RPV[PS_table$Pop=="HZIL"],
        PS_table$RPV[PS_table$Pop=="REIL"], PS_table$RPV[PS_table$Pop=="TYIL"],
        PS_table$RPV[PS_table$Pop=="ZYIL"], PS_table$RPV[PS_table$Pop=="GMIL"])])
 axis(side = 2, line=-2.8, lty=0)

axis(side = 2, line=-2, tck=.9, bg="gray85", labels = F)
       
axis(side = 1, at=c(seq(from=1, to=10.9, by=.9)), tick = F, line=-.8, labels=c( 
                                "France\n(AFFR)", "Spain\n(DEES)", 
                                "Spain\n(HCES)", "Spain\n(HMES)", 
                                "Spain\n(IMES)", "Spain\n(MAES)", 
                                "Israel\n(GHIL)", "Israel\n(HZIL)", 
                                "Israel\n(REIL)", "Israel\n(TYIL)",
                                "Israel\n(ZYIL)", "Israel\n(GMIL)"))



par(cex=.9)

axis(side=3, at=3.5, labels="Western", cex=2, tick=F, line=-3)
axis(side=3, at=8.5, labels="Eastern", cex=2, tick=F, line=-3)

axis(side=3, at=10.7, labels=expression(italic("R. pugioniformis")), cex=2, tick=F, line=-1)
axis(side=3, at=6, labels=expression(paste(italic("R.r. raphanistrum")," inside native range")), cex=1.2, tick=F, line=-1)

lines(c(6.1,6.1), c(50,150), lty=3)
lines(c(10.5,10.5), c(25,175), lty=6)

# Plot weeds

par(fig=c(0.1,.8,.37,.70), new=TRUE, cex=.7)

plot(weed.data$GTF ~ weed.data$Order, add=T, frame=F, axes=F, ylim=c(0,200), pch=16,
        bg=hmcols[c(
        PS_table$RPV[PS_table$Pop=="AUFI"], PS_table$RPV[PS_table$Pop=="BINY"],
        PS_table$RPV[PS_table$Pop=="COAU"], PS_table$RPV[PS_table$Pop=="KAMI"], 
        PS_table$RPV[PS_table$Pop=="MAFI"], PS_table$RPV[PS_table$Pop=="NAAU"],
        PS_table$RPV[PS_table$Pop=="WEAU"])],
        varwidth=TRUE)
axis(side = 2, line=-1, tck=.9, bg="gray85", yaxp=c(0, 150, 3))

axis(side = 1, at=c(1:7), tick = F, line=-1,
     label=c("Finland\n(AUFI)", "New York\n(BINY)",
             "Australia 1\n(COAU)", "Michigan\n(KAMI)",
             "Finland\n(MAFI)", "Australia 3\n(NAAU)",
             "Australia 2\n(WEAU)" ))


par(las=3)
mtext(text="Flowering Time in Days", side = 2, adj = 0, line = 2 )
par(las=1)

par(cex=1)
axis(side = 3, at=4.5, tick = F, line=-5, labels=expression(paste(italic("R.r. raphanistrum")," outside native range")))

par(fig=c(0.8,1,.4,.65), new=TRUE, cex=.7)


gradient.rect(4.5, 2, 5.5, 130, col=hmcols, gradient = "vertical", border=NA)
text(3, 150, labels = "Flowered without\ncold treatment", cex=1.2)
text(2.5, 10, labels="0% of population")
text(2.5, 120, labels="100% of population")
polygon(c(.4,6,6,.4), c(0,0,170,170))


par(fig=c(0.1,1,.03,.31), new=TRUE, cex=.7)

par(new=TRUE, cex=.7)

plot(crop.data$GTF ~ crop.data$Order, add=T, frame=F, axes=F, ylim=c(0,200), pch=16,
         wex=.7, at=c(seq(from=.9, to=18.9, by=.9)), bg=hmcols[c(
        PS_table$RPV[PS_table$Pop=="CGBC"], PS_table$RPV[PS_table$Pop=="FGBC"],
        PS_table$RPV[PS_table$Pop=="MYJO"], PS_table$RPV[PS_table$Pop=="NEJS"],
        PS_table$RPV[PS_table$Pop=="TOBG"], PS_table$RPV[PS_table$Pop=="WMBG"],
        PS_table$RPV[PS_table$Pop=="CBBG"], PS_table$RPV[PS_table$Pop=="DAJO"],
        PS_table$RPV[PS_table$Pop=="ESNK"], PS_table$RPV[PS_table$Pop=="FRSI"],
        PS_table$RPV[PS_table$Pop=="LBBC"], PS_table$RPV[PS_table$Pop=="RABS"], 
        PS_table$RPV[PS_table$Pop=="RBBC"], PS_table$RPV[PS_table$Pop=="SPNK"], 
        PS_table$RPV[PS_table$Pop=="ADOL"], PS_table$RPV[PS_table$Pop=="AROL"], 
        PS_table$RPV[PS_table$Pop=="COOL"], PS_table$RPV[PS_table$Pop=="OIBG"], 
        PS_table$RPV[PS_table$Pop=="MABG"], PS_table$RPV[PS_table$Pop=="RABG"], 
        PS_table$RPV[PS_table$Pop=="RAJS"])])
axis(side = 2, line=-1, tck=.95, bg='gray85')

axis(side = 1, at=c(seq(from=.9, to=19, by=1.8)), tick = F, line=-1.5, ylim=c(0, 300),
     label=c("CGBC","MYJO","TOBG","CBBG","ESNK",
             "LBBC","RBBC","ADOL","COOL","MABG","RAJS"))

axis(side = 1, at=c(seq(from=1.8, to=19, by=1.8)), tick = F, line=-1.5, ylim=c(0, 300),
     label=c("FGBC","NEJS","WMBG","DAJO","FRSI",
             "RABS","SPNK","AROL","OIBG","RABG"))

par(cex=1)
axis(side = 3, at=9.5, tick = F, line=-1, labels=expression(italic("R. sativus")))

par(cex=.9)
axis(side=3, at=2.7, tick=F, labels = "Daikon", line=-4)
axis(side=3, at=9.5, tick=F, labels="European", line=-4)
axis(side=3, at=15, tick=F, labels="Oilseed", line=-4)
axis(side=3, at=18, tick=F, labels="Rattail", line=-4)


lines(c(5.85,5.85), c(25,175), lty=6)
lines(c(13.1,13.1), c(25,175), lty=6)
lines(c(16.7,16.7), c(25,175), lty=6)




```

```{r ExperimentalVar, echo=FALSE, fig.path="../Figures/", fig.keep="all", message=FALSE, warning=FALSE, fig.height=10, fig.width=14, warning=FALSE}

group_by(FlowData, Pop) %>%
ggplot( aes(Pop:Experiment, SurviveGTF, col=Experiment)) +
  geom_boxplot() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) 

#group_by(RawData, Pop) %>%
#ggplot( aes(Pop:Experiment, GTF, col=Experiment, group=Pop)) +
#  geom_boxplot() +
#  theme(axis.text.x = element_text(angle = 90, hjust = 1)) 

group_by(FlowData, Pop) %>%
ggplot( aes(Experiment, SurviveGTF, col=Experiment)) +
  geom_boxplot() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  facet_wrap("Pop")

```

```{r SmallExperimentalVar, echo=FALSE, fig.path="../Figures/", fig.keep="all", message=FALSE, warning=FALSE, fig.height=10, fig.width=14, warning=FALSE}

filter(FlowData, Pop=="BINY"| Pop=="MAES"| Pop=="SAES") %>%
  filter(!is.na(BoltFlower)) %>%
group_by( Pop) %>%
ggplot( aes(Experiment, BoltFlower, col=Experiment)) +
  geom_jitter(width = .2) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 14),
        axis.text.y = element_text(size = 14),
        legend.title = element_text( size=18, face="bold"),
        legend.text = element_text( size = 16),
        strip.text =  element_text( size = 14)) +
  facet_wrap("Pop", scales = "free_x") +
  guides(colour = guide_legend(override.aes = list(size=8)))

```

Testing some ANOVA models


```{r, echo=FALSE, warning=FALSE}

test1 <- lm(RawData$GTF ~ RawData$Experiment + RawData$Experiment:RawData$Pop)


test2 <- lm(data = RawData, formula = SurviveGTF ~ Experiment + Seedstock + msDaysVernNoNA + Experiment:Pop)

car::Anova(test1, test.statistic="F")
car::Anova(test2, test.statistic="F")


```

